<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Iliana Papadopoulou" />

<meta name="date" content="2025-05-11" />

<title>Alveolar Macrophages Pathway Enrichment Analysis</title>

<script src="site_libs/header-attrs-2.24/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Alveolar_Mph_mouse_lung_PH</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Alveolar Macrophages Pathway Enrichment
Analysis</h1>
<h4 class="author">Iliana Papadopoulou</h4>
<h4 class="date">2025-05-11</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2025-05-11
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong>
<code>Alveolar_Mph_mouse_lung_PH/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20250507code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20250507)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20250507code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20250507)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomilianapapAlveolarMphmouselungPHtree631ba323f355a21c4b0f3958ac4afc3d1ca4c37dtargetblank631ba32a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/iliana-pap/Alveolar_Mph_mouse_lung_PH/tree/631ba323f355a21c4b0f3958ac4afc3d1ca4c37d" target="_blank">631ba32</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomilianapapAlveolarMphmouselungPHtree631ba323f355a21c4b0f3958ac4afc3d1ca4c37dtargetblank631ba32a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/iliana-pap/Alveolar_Mph_mouse_lung_PH/tree/631ba323f355a21c4b0f3958ac4afc3d1ca4c37d" target="_blank">631ba32</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    data/Output/

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown
(<code>analysis/Alveolar_Macrophages_GSEA.Rmd</code>) and HTML
(<code>docs/Alveolar_Macrophages_GSEA.html</code>) files. If you’ve
configured a remote Git repository (see <code>?wflow_git_remote</code>),
click on the hyperlinks in the table below to view the files as they
were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/iliana-pap/Alveolar_Mph_mouse_lung_PH/blob/631ba323f355a21c4b0f3958ac4afc3d1ca4c37d/analysis/Alveolar_Macrophages_GSEA.Rmd" target="_blank">631ba32</a>
</td>
<td>
iliana-pap
</td>
<td>
2025-05-11
</td>
<td>
update pathway analysis
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/iliana-pap/Alveolar_Mph_mouse_lung_PH/3692a1165fe546e4d4dc3de219868234b140744f/docs/Alveolar_Macrophages_GSEA.html" target="_blank">3692a11</a>
</td>
<td>
iliana-pap
</td>
<td>
2025-05-10
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/iliana-pap/Alveolar_Mph_mouse_lung_PH/blob/084a4623beeac1ad3db43f180358f92baf497f80/analysis/Alveolar_Macrophages_GSEA.Rmd" target="_blank">084a462</a>
</td>
<td>
iliana-pap
</td>
<td>
2025-05-10
</td>
<td>
perform pathway analysis in alveolar macrophages per condition
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="customize-parameters" class="section level2">
<h2>Customize parameters</h2>
<pre class="r"><code>library(plyr)
library(biomaRt)
library(dplyr)
library(tidyverse)
library(simplifyEnrichment)
library(msigdbr)
library(msigdbr)
library(here)
library(stats)
library(ggsci)
library(clusterProfiler)
library(dict)
library(org.Mm.eg.db)
library(DOSE)
library(Seurat)
library(openxlsx)
library(zeallot)
library(ggsci)
library(openxlsx)
library(pheatmap)
library(stringr)
library(viridis)
library(scCustomize)
library(stringr)
library(gridExtra)
library(grid)

httr::set_config(httr::config(ssl_verifypeer = FALSE))

organism &lt;- &quot;org.Mm.eg.db&quot;
disease_phase &lt;- &quot;&quot;
datatype &lt;- &#39;SYMBOL&#39;</code></pre>
</div>
<div
id="pathway-enrichment-analysis-of-control-vs-ph-chfpef-mice-in-alveolar-macrophages"
class="section level1">
<h1><strong>4.Pathway enrichment analysis of control vs PH-cHFpEF mice
in alveolar macrophages</strong></h1>
<div id="set-directory" class="section level2">
<h2>Set directory</h2>
<pre class="r"><code>basedir &lt;- here()
source(paste0(basedir,&#39;/analysis/Lung_snRNAseq_functions.R&#39;), echo=FALSE)

# Create directory to save preprocessed seurat objects
output_obj_dir &lt;- paste0(basedir,&quot;/data/Output/&quot;)
dir.create(output_obj_dir)</code></pre>
</div>
<div id="create-directory-for-saving-clustering-results"
class="section level2">
<h2>Create directory for saving clustering results</h2>
<pre class="r"><code>GSEADir &lt;- paste0(output_obj_dir,&quot;GSEA&quot;)
dir.create(GSEADir)</code></pre>
</div>
<div id="read-seurat-annotated-object" class="section level2">
<h2>Read seurat annotated object</h2>
<pre class="r"><code>LungMu_obj &lt;- readRDS(file= paste0(output_obj_dir, &quot;340821_Mu_Lung_snRNAseq_merged_annotated.rds&quot; ))

LungMu_obj$celltype_condition &lt;- paste0(LungMu_obj$label,&quot;_&quot;,LungMu_obj$condition)</code></pre>
</div>
<div id="color-vectors---define-color-palete" class="section level2">
<h2>Color vectors - Define color palete</h2>
<pre class="r"><code>colCond &lt;- c(&quot;#631879FF&quot;,&quot;#F16913&quot;)
names(colCond) &lt;- unique(LungMu_obj$condition)</code></pre>
<pre class="r"><code>GSEA_Analysis &lt;- function(data,organism,datatype,disease_phase,celltype,outdir){
  obj_sub &lt;- subset(data, label == celltype)
  print(unique(obj_sub$label))
  Idents(obj_sub) &lt;- obj_sub$condition
  markers &lt;- FindAllMarkers(object = obj_sub, 
                            only.pos = TRUE,
                            logfc.threshold = 0.25,
                            min.pct = 0.1,
                            test.use = &quot;wilcox&quot;)  
  
  Vec &lt;- sort(unique(obj_sub$condition))
  EnrichParameters &lt;-customize_parameters(Vec,markers,organism,datatype,disease_phase, GSEADir)
  
  for (i in seq(1,length(EnrichParameters$enrichcl_list))){
    terms&lt;- EnrichParameters$enrichcl_list[[i]]@result
    condition &lt;- Vec[i]
    condition &lt;- gsub(&quot;/&quot;, &quot;_&quot;, condition)
    celltype &lt;- gsub(&quot;/&quot;, &quot;_&quot;, celltype)
    write.xlsx(terms, paste0(outdir,&quot;/&quot;,&quot;GO_Pathways_&quot;,celltype,&quot;_&quot;,condition,&quot;.xlsx&quot;),rowNames = TRUE)
  } 
  return(EnrichParameters)
}


celltype &lt;-&quot;Alveolar Mph&quot;
data &lt;- LungMu_obj
EnrichParameters &lt;- GSEA_Analysis(data,organism,datatype,disease_phase,celltype,GSEADir)</code></pre>
<pre><code>[1] &quot;Alveolar Mph&quot;
[1] &quot;End Enrichment analysis ...&quot;
[1] &quot;End Enrichment analysis ...&quot;</code></pre>
</div>
<div
id="join-files-in-order-to-gather-all-info-in-one-file---for-gene-name"
class="section level2">
<h2>Join files in order to gather all info in one file - for gene
name</h2>
<pre class="r"><code># For control
GO_semantic_Al_Mph_CD &lt;- read.xlsx(paste0(GSEADir,&quot;/&quot;,&quot;GO_Pathways_Alveolar Mph_Control.xlsx&quot;))
GO_semantic_Al_Mph_CD[,1] &lt;- NULL

# For PH-cHFpEF
GO_semantic_HFD &lt;- read.xlsx(paste0(GSEADir,&quot;/&quot;,&quot;GO_Pathways_Alveolar Mph_PH-cHFpEF.xlsx&quot;))
GO_semantic_HFD[,1] &lt;- NULL</code></pre>
</div>
<div
id="find-overlapping-pathways-between-alveolar-macrophages-control_vs_ph-chfpef-vs-control"
class="section level2">
<h2>Find overlapping pathways between Alveolar macrophages
control_vs_PH-cHFpEF vs control</h2>
<pre class="r"><code>df_overlapping_pathways &lt;- inner_join(GO_semantic_Al_Mph_CD, GO_semantic_HFD,by=c(&quot;ID&quot;,&quot;Description&quot;) )
overlapping_pathways &lt;- df_overlapping_pathways$ID</code></pre>
</div>
<div
id="find-exclusive-pathways-in-both-alveolar-macrophages-control_vs_ph-chfpef-vs-control"
class="section level2">
<h2>Find exclusive pathways in both Alveolar macrophages
control_vs_PH-cHFpEF vs control</h2>
<pre class="r"><code>df_list &lt;-list(GO_semantic_Al_Mph_CD, GO_semantic_HFD)
GO_semantic_AlvMph&lt;-do.call(&quot;rbind&quot;, df_list)
GO_semantic_AlvMph &lt;- GO_semantic_AlvMph %&gt;%
  filter (! ID %in% overlapping_pathways)</code></pre>
</div>
<div
id="figure-2e---plot-selected-overlapping-pathways-and-exclusive-between-alveolar-macrophages-control-vs-ph-chfpef-mice"
class="section level2">
<h2>Figure 2E - Plot selected overlapping pathways and exclusive between
Alveolar macrophages control vs PH-cHFpEF mice</h2>
<pre class="r"><code>pathways &lt;- c(
&quot;GO:0050881&quot;,&quot;GO:0086001&quot;,  &quot;GO:0098760&quot;, &quot;GO:0098779&quot;,
&quot;GO:0043277&quot;,&quot;GO:0031589&quot;,&quot;GO:0030509&quot;,&quot;GO:1903036&quot;, &quot;GO:0071353&quot;,&quot;GO:2000343&quot;,&quot;GO:0071248&quot;
)


t_df_overlapping_pathways &lt;- df_overlapping_pathways
PATH_per_cond &lt;- t_df_overlapping_pathways[,c(&quot;ID&quot;,&quot;Description&quot;, &quot;p.adjust.x&quot;, &quot;analysis.x&quot;, &quot;p.adjust.y&quot;, &quot;analysis.y&quot;)]

# For pathways that are shared between conditions
PATH_per_cond &lt;- PATH_per_cond %&gt;% filter(ID %in% pathways, .keep_all = TRUE)

rownames(PATH_per_cond) &lt;- PATH_per_cond$Description
PATH_per_cond$Description &lt;- NULL
PATH_per_cond$ID &lt;- NULL

colnames(PATH_per_cond) &lt;- c(unique(PATH_per_cond$analysis.x), 
                             &quot;analysis.x&quot;,
                            unique(PATH_per_cond$analysis.y),
                             &quot;analysis.y&quot;)
PATH_per_cond$analysis.x &lt;- NULL
PATH_per_cond$analysis.y &lt;- NULL

# For pathways that are exclusive in each condition
exl_path &lt;- setdiff(pathways, PATH_per_cond$ID)
exl_path_cond &lt;- GO_semantic_AlvMph[,c(&quot;ID&quot;,&quot;Description&quot;, &quot;p.adjust&quot;, &quot;analysis&quot;)]
exl_path_cond &lt;- exl_path_cond %&gt;% filter(ID %in% exl_path, .keep_all = TRUE)
exl_path_cond &lt;- exl_path_cond %&gt;%
  pivot_wider(id_cols = c(&#39;ID&#39;,&#39;Description&#39;), names_from = &#39;analysis&#39;,
              values_from = &#39;p.adjust&#39;) %&gt;%
  as.data.frame()
rownames(exl_path_cond) &lt;- exl_path_cond$Description
exl_path_cond$Description &lt;- NULL
exl_path_cond$ID &lt;- NULL

 if ((&quot;Control&quot; %in% colnames(exl_path_cond) == &quot;FALSE&quot;)){
   exl_path_cond$Control &lt;- NA
 }

# Combine two dfs
PATH_per_cond &lt;- rbind(PATH_per_cond, exl_path_cond)

# Log the dataframe
PATH_per_cond[] &lt;- lapply(PATH_per_cond, function(x) -1 *(log10(x)))
PATH_per_cond[is.na(PATH_per_cond)] &lt;- 0

PATH_per_cond &lt;- PATH_per_cond %&gt;% 
  mutate(across(where(is.numeric))/rowSums(across(where(is.numeric))))

# Generate annotations for columns
annotation_col = data.frame(
                    condition = colnames(PATH_per_cond)
                )
rownames(annotation_col) = colnames(PATH_per_cond)
ann_colors = list(condition = colCond)

# Choose color palette
palette_length = 100
     
pheatmap(PATH_per_cond, cluster_rows=T, cluster_cols=F, cellwidth = 10, cellheight = 10, scale = &quot;none&quot;,
        color = colorRampPalette(RColorBrewer::brewer.pal(n = 11, name = &quot;RdYlBu&quot;) )(50)|&gt; rev() ,
        annotation_col = annotation_col, annotation_colors = ann_colors) </code></pre>
<p><img src="figure/Alveolar_Macrophages_GSEA.Rmd/visualite%20pathways-1.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-visualite-pathways-1">
Past versions of "visualite pathways-1.png"
</button>
</p>
<div id="fig-no-spaces-visualite-pathways-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/iliana-pap/Alveolar_Mph_mouse_lung_PH/blob/3692a1165fe546e4d4dc3de219868234b140744f/docs/figure/Alveolar_Macrophages_GSEA.Rmd/visualite pathways-1.png" target="_blank">3692a11</a>
</td>
<td>
iliana-pap
</td>
<td>
2025-05-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="session-information" class="section level2">
<h2>Session Information</h2>
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.3.1 (2023-06-16)
Platform: aarch64-apple-darwin20 (64-bit)
Running under: macOS 15.3.2

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRblas.0.dylib 
LAPACK: /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.11.0

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

time zone: Europe/Zurich
tzcode source: internal

attached base packages:
[1] stats4    grid      stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] gridExtra_2.3             scCustomize_1.1.3        
 [3] viridis_0.6.4             viridisLite_0.4.2        
 [5] pheatmap_1.0.12           zeallot_0.1.0            
 [7] openxlsx_4.2.5.2          SeuratObject_5.0.2       
 [9] Seurat_4.3.0.1            DOSE_3.26.1              
[11] org.Mm.eg.db_3.17.0       AnnotationDbi_1.62.2     
[13] IRanges_2.34.1            S4Vectors_0.38.1         
[15] Biobase_2.60.0            dict_0.10.0              
[17] clusterProfiler_4.8.3     ggsci_3.0.0              
[19] here_1.0.1                msigdbr_7.5.1            
[21] simplifyEnrichment_1.10.0 BiocGenerics_0.46.0      
[23] lubridate_1.9.4           forcats_1.0.0            
[25] stringr_1.5.0             purrr_1.0.2              
[27] readr_2.1.4               tidyr_1.3.1              
[29] tibble_3.2.1              ggplot2_3.5.2            
[31] tidyverse_2.0.0           dplyr_1.1.4              
[33] biomaRt_2.56.1            plyr_1.8.9               
[35] workflowr_1.7.1          

loaded via a namespace (and not attached):
  [1] spatstat.sparse_3.0-2   fs_1.6.3                matrixStats_1.0.0      
  [4] bitops_1.0-7            enrichplot_1.20.1       HDO.db_0.99.1          
  [7] httr_1.4.6              RColorBrewer_1.1-3      doParallel_1.0.17      
 [10] sctransform_0.3.5       tools_4.3.1             utf8_1.2.3             
 [13] R6_2.5.1                uwot_0.1.16             lazyeval_0.2.2         
 [16] GetoptLong_1.0.5        withr_3.0.2             sp_2.1-4               
 [19] prettyunits_1.1.1       progressr_0.14.0        cli_3.6.3              
 [22] spatstat.explore_3.2-1  scatterpie_0.2.1        slam_0.1-50            
 [25] sass_0.4.7              spatstat.data_3.0-1     tm_0.7-11              
 [28] pbapply_1.7-2           ggridges_0.5.4          yulab.utils_0.0.9      
 [31] gson_0.1.0              parallelly_1.37.1       limma_3.56.2           
 [34] rstudioapi_0.15.0       RSQLite_2.3.1           generics_0.1.3         
 [37] gridGraphics_0.5-1      shape_1.4.6             spatstat.random_3.1-5  
 [40] ica_1.0-3               zip_2.3.0               GO.db_3.17.0           
 [43] Matrix_1.6-4            ggbeeswarm_0.7.2        fansi_1.0.4            
 [46] abind_1.4-5             lifecycle_1.0.4         whisker_0.4.1          
 [49] yaml_2.3.7              snakecase_0.11.0        qvalue_2.32.0          
 [52] BiocFileCache_2.8.0     Rtsne_0.16              paletteer_1.6.0        
 [55] blob_1.2.4              promises_1.2.1          crayon_1.5.3           
 [58] miniUI_0.1.1.1          lattice_0.21-8          cowplot_1.1.1          
 [61] KEGGREST_1.40.0         pillar_1.9.0            knitr_1.43             
 [64] ComplexHeatmap_2.16.0   fgsea_1.26.0            rjson_0.2.21           
 [67] future.apply_1.11.2     codetools_0.2-19        fastmatch_1.1-4        
 [70] leiden_0.4.3            glue_1.8.0              getPass_0.2-4          
 [73] downloader_0.4          ggfun_0.1.2             data.table_1.14.8      
 [76] vctrs_0.6.5             png_0.1-8               treeio_1.24.3          
 [79] spam_2.10-0             gtable_0.3.3            rematch2_2.1.2         
 [82] cachem_1.0.8            xfun_0.40               mime_0.12              
 [85] tidygraph_1.2.3         survival_3.5-5          iterators_1.0.14       
 [88] ellipsis_0.3.2          fitdistrplus_1.1-11     ROCR_1.0-11            
 [91] nlme_3.1-162            ggtree_3.8.2            bit64_4.0.5            
 [94] progress_1.2.2          filelock_1.0.2          RcppAnnoy_0.0.21       
 [97] GenomeInfoDb_1.36.2     rprojroot_2.0.3         bslib_0.5.1            
[100] irlba_2.3.5.1           vipor_0.4.5             KernSmooth_2.23-21     
[103] colorspace_2.1-0        DBI_1.1.3               ggrastr_1.0.2          
[106] tidyselect_1.2.0        processx_3.8.2          proxyC_0.3.3           
[109] bit_4.0.5               compiler_4.3.1          curl_5.0.2             
[112] git2r_0.36.2            xml2_1.3.5              NLP_0.2-1              
[115] plotly_4.10.2           shadowtext_0.1.2        scales_1.3.0           
[118] lmtest_0.9-40           callr_3.7.3             rappdirs_0.3.3         
[121] goftest_1.2-3           digest_0.6.36           spatstat.utils_3.0-5   
[124] rmarkdown_2.24          XVector_0.40.0          htmltools_0.5.6        
[127] pkgconfig_2.0.3         highr_0.10              dbplyr_2.3.3           
[130] fastmap_1.1.1           htmlwidgets_1.6.2       rlang_1.1.4            
[133] GlobalOptions_0.1.2     shiny_1.7.5             farver_2.1.1           
[136] jquerylib_0.1.4         zoo_1.8-12              jsonlite_1.8.7         
[139] BiocParallel_1.34.2     GOSemSim_2.26.1         RCurl_1.98-1.12        
[142] magrittr_2.0.3          GenomeInfoDbData_1.2.10 ggplotify_0.1.2        
[145] dotCall64_1.1-1         patchwork_1.3.0         munsell_0.5.0          
[148] Rcpp_1.0.13             ape_5.7-1               babelgene_22.9         
[151] reticulate_1.31         stringi_1.7.12          ggraph_2.1.0           
[154] zlibbioc_1.46.0         MASS_7.3-60             org.Hs.eg.db_3.17.0    
[157] parallel_4.3.1          listenv_0.9.1           ggrepel_0.9.6          
[160] deldir_1.0-9            Biostrings_2.68.1       graphlayouts_1.0.0     
[163] splines_4.3.1           tensor_1.5              hms_1.1.3              
[166] circlize_0.4.15         ps_1.7.5                igraph_1.5.1           
[169] spatstat.geom_3.2-4     reshape2_1.4.4          XML_3.99-0.14          
[172] evaluate_0.21           RcppParallel_5.1.7      ggprism_1.0.4          
[175] tzdb_0.4.0              foreach_1.5.2           tweenr_2.0.2           
[178] httpuv_1.6.11           RANN_2.6.1              polyclip_1.10-4        
[181] scattermore_1.2         future_1.33.2           clue_0.3-64            
[184] ggforce_0.4.1           janitor_2.2.0           xtable_1.8-4           
[187] tidytree_0.4.5          later_1.3.1             aplot_0.2.0            
[190] beeswarm_0.4.0          memoise_2.0.1           cluster_2.1.4          
[193] timechange_0.3.0        globals_0.16.3         </code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.3.1 (2023-06-16)
Platform: aarch64-apple-darwin20 (64-bit)
Running under: macOS 15.3.2

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRblas.0.dylib 
LAPACK: /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.11.0

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

time zone: Europe/Zurich
tzcode source: internal

attached base packages:
[1] stats4    grid      stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] gridExtra_2.3             scCustomize_1.1.3        
 [3] viridis_0.6.4             viridisLite_0.4.2        
 [5] pheatmap_1.0.12           zeallot_0.1.0            
 [7] openxlsx_4.2.5.2          SeuratObject_5.0.2       
 [9] Seurat_4.3.0.1            DOSE_3.26.1              
[11] org.Mm.eg.db_3.17.0       AnnotationDbi_1.62.2     
[13] IRanges_2.34.1            S4Vectors_0.38.1         
[15] Biobase_2.60.0            dict_0.10.0              
[17] clusterProfiler_4.8.3     ggsci_3.0.0              
[19] here_1.0.1                msigdbr_7.5.1            
[21] simplifyEnrichment_1.10.0 BiocGenerics_0.46.0      
[23] lubridate_1.9.4           forcats_1.0.0            
[25] stringr_1.5.0             purrr_1.0.2              
[27] readr_2.1.4               tidyr_1.3.1              
[29] tibble_3.2.1              ggplot2_3.5.2            
[31] tidyverse_2.0.0           dplyr_1.1.4              
[33] biomaRt_2.56.1            plyr_1.8.9               
[35] workflowr_1.7.1          

loaded via a namespace (and not attached):
  [1] spatstat.sparse_3.0-2   fs_1.6.3                matrixStats_1.0.0      
  [4] bitops_1.0-7            enrichplot_1.20.1       HDO.db_0.99.1          
  [7] httr_1.4.6              RColorBrewer_1.1-3      doParallel_1.0.17      
 [10] sctransform_0.3.5       tools_4.3.1             utf8_1.2.3             
 [13] R6_2.5.1                uwot_0.1.16             lazyeval_0.2.2         
 [16] GetoptLong_1.0.5        withr_3.0.2             sp_2.1-4               
 [19] prettyunits_1.1.1       progressr_0.14.0        cli_3.6.3              
 [22] spatstat.explore_3.2-1  scatterpie_0.2.1        slam_0.1-50            
 [25] sass_0.4.7              spatstat.data_3.0-1     tm_0.7-11              
 [28] pbapply_1.7-2           ggridges_0.5.4          yulab.utils_0.0.9      
 [31] gson_0.1.0              parallelly_1.37.1       limma_3.56.2           
 [34] rstudioapi_0.15.0       RSQLite_2.3.1           generics_0.1.3         
 [37] gridGraphics_0.5-1      shape_1.4.6             spatstat.random_3.1-5  
 [40] ica_1.0-3               zip_2.3.0               GO.db_3.17.0           
 [43] Matrix_1.6-4            ggbeeswarm_0.7.2        fansi_1.0.4            
 [46] abind_1.4-5             lifecycle_1.0.4         whisker_0.4.1          
 [49] yaml_2.3.7              snakecase_0.11.0        qvalue_2.32.0          
 [52] BiocFileCache_2.8.0     Rtsne_0.16              paletteer_1.6.0        
 [55] blob_1.2.4              promises_1.2.1          crayon_1.5.3           
 [58] miniUI_0.1.1.1          lattice_0.21-8          cowplot_1.1.1          
 [61] KEGGREST_1.40.0         pillar_1.9.0            knitr_1.43             
 [64] ComplexHeatmap_2.16.0   fgsea_1.26.0            rjson_0.2.21           
 [67] future.apply_1.11.2     codetools_0.2-19        fastmatch_1.1-4        
 [70] leiden_0.4.3            glue_1.8.0              getPass_0.2-4          
 [73] downloader_0.4          ggfun_0.1.2             data.table_1.14.8      
 [76] vctrs_0.6.5             png_0.1-8               treeio_1.24.3          
 [79] spam_2.10-0             gtable_0.3.3            rematch2_2.1.2         
 [82] cachem_1.0.8            xfun_0.40               mime_0.12              
 [85] tidygraph_1.2.3         survival_3.5-5          iterators_1.0.14       
 [88] ellipsis_0.3.2          fitdistrplus_1.1-11     ROCR_1.0-11            
 [91] nlme_3.1-162            ggtree_3.8.2            bit64_4.0.5            
 [94] progress_1.2.2          filelock_1.0.2          RcppAnnoy_0.0.21       
 [97] GenomeInfoDb_1.36.2     rprojroot_2.0.3         bslib_0.5.1            
[100] irlba_2.3.5.1           vipor_0.4.5             KernSmooth_2.23-21     
[103] colorspace_2.1-0        DBI_1.1.3               ggrastr_1.0.2          
[106] tidyselect_1.2.0        processx_3.8.2          proxyC_0.3.3           
[109] bit_4.0.5               compiler_4.3.1          curl_5.0.2             
[112] git2r_0.36.2            xml2_1.3.5              NLP_0.2-1              
[115] plotly_4.10.2           shadowtext_0.1.2        scales_1.3.0           
[118] lmtest_0.9-40           callr_3.7.3             rappdirs_0.3.3         
[121] goftest_1.2-3           digest_0.6.36           spatstat.utils_3.0-5   
[124] rmarkdown_2.24          XVector_0.40.0          htmltools_0.5.6        
[127] pkgconfig_2.0.3         highr_0.10              dbplyr_2.3.3           
[130] fastmap_1.1.1           htmlwidgets_1.6.2       rlang_1.1.4            
[133] GlobalOptions_0.1.2     shiny_1.7.5             farver_2.1.1           
[136] jquerylib_0.1.4         zoo_1.8-12              jsonlite_1.8.7         
[139] BiocParallel_1.34.2     GOSemSim_2.26.1         RCurl_1.98-1.12        
[142] magrittr_2.0.3          GenomeInfoDbData_1.2.10 ggplotify_0.1.2        
[145] dotCall64_1.1-1         patchwork_1.3.0         munsell_0.5.0          
[148] Rcpp_1.0.13             ape_5.7-1               babelgene_22.9         
[151] reticulate_1.31         stringi_1.7.12          ggraph_2.1.0           
[154] zlibbioc_1.46.0         MASS_7.3-60             org.Hs.eg.db_3.17.0    
[157] parallel_4.3.1          listenv_0.9.1           ggrepel_0.9.6          
[160] deldir_1.0-9            Biostrings_2.68.1       graphlayouts_1.0.0     
[163] splines_4.3.1           tensor_1.5              hms_1.1.3              
[166] circlize_0.4.15         ps_1.7.5                igraph_1.5.1           
[169] spatstat.geom_3.2-4     reshape2_1.4.4          XML_3.99-0.14          
[172] evaluate_0.21           RcppParallel_5.1.7      ggprism_1.0.4          
[175] tzdb_0.4.0              foreach_1.5.2           tweenr_2.0.2           
[178] httpuv_1.6.11           RANN_2.6.1              polyclip_1.10-4        
[181] scattermore_1.2         future_1.33.2           clue_0.3-64            
[184] ggforce_0.4.1           janitor_2.2.0           xtable_1.8-4           
[187] tidytree_0.4.5          later_1.3.1             aplot_0.2.0            
[190] beeswarm_0.4.0          memoise_2.0.1           cluster_2.1.4          
[193] timechange_0.3.0        globals_0.16.3         </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
