<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Iliana Papadopoulou" />

<meta name="date" content="2025-05-10" />

<title>Alveolar_Macrophages_analysis</title>

<script src="site_libs/header-attrs-2.24/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Alveolar_Mph_mouse_lung_PH</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Alveolar_Macrophages_analysis</h1>
<h4 class="author">Iliana Papadopoulou</h4>
<h4 class="date">2025-05-10</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2025-05-10
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong>
<code>Alveolar_Mph_mouse_lung_PH/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20250507code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20250507)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20250507code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20250507)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomilianapapAlveolarMphmouselungPHtree12dfc8b5ea63202cb2263b46f6d1fa7deb7170f3targetblank12dfc8ba">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/iliana-pap/Alveolar_Mph_mouse_lung_PH/tree/12dfc8b5ea63202cb2263b46f6d1fa7deb7170f3" target="_blank">12dfc8b</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomilianapapAlveolarMphmouselungPHtree12dfc8b5ea63202cb2263b46f6d1fa7deb7170f3targetblank12dfc8ba"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/iliana-pap/Alveolar_Mph_mouse_lung_PH/tree/12dfc8b5ea63202cb2263b46f6d1fa7deb7170f3" target="_blank">12dfc8b</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    data/Output/

Untracked files:
    Untracked:  analysis/Lung_snRNAseq_functions.R

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/Alveolar_Mph_analysis.Rmd</code>)
and HTML (<code>docs/Alveolar_Mph_analysis.html</code>) files. If you’ve
configured a remote Git repository (see <code>?wflow_git_remote</code>),
click on the hyperlinks in the table below to view the files as they
were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/iliana-pap/Alveolar_Mph_mouse_lung_PH/blob/12dfc8b5ea63202cb2263b46f6d1fa7deb7170f3/analysis/Alveolar_Mph_analysis.Rmd" target="_blank">12dfc8b</a>
</td>
<td>
iliana-pap
</td>
<td>
2025-05-10
</td>
<td>
analyze alveolar macrophages phenotype
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div
id="analysis-of-alveolar-macrophages-extracted-from-murine-lung-samples-of-healthy-mice-vs-mice-with-pulmonary-hypertension-and-hfpef"
class="section level1">
<h1>3. Analysis of alveolar macrophages extracted from murine lung
samples of healthy mice vs mice with pulmonary hypertension and
HFpEF</h1>
<div id="load-r-libraries" class="section level2">
<h2>Load R libraries</h2>
<pre class="r"><code>suppressPackageStartupMessages({
  library(plyr)
  library(patchwork)
  library(Seurat)
  library(stringr)
  library(dplyr)
  library(tidyverse)
  library(here)
  library(openxlsx)
  library(zeallot)
  library(ggsci)
  library(openxlsx)
  library(pheatmap)
  library(stringr)
  library(viridis)
  library(scCustomize)
  library(stringr)
  library(gridExtra)
  library(grid)
  library(tidyverse)
  library(ggsignif)
  library(ggbeeswarm)
  library(ggrepel)
  library(ggpubr)
})</code></pre>
</div>
<div id="set-directory" class="section level2">
<h2>Set directory</h2>
<pre class="r"><code>basedir &lt;- here()
source(paste0(basedir,&#39;/analysis/Lung_snRNAseq_functions.R&#39;), echo=FALSE)

# Create directory to save preprocessed seurat objects
output_obj_dir &lt;- paste0(basedir,&quot;/data/Output/&quot;)
dir.create(output_obj_dir)</code></pre>
</div>
<div id="read-annotated-murine-lung-seurat-object"
class="section level2">
<h2>Read annotated murine lung seurat object</h2>
<pre class="r"><code>LungMu_obj &lt;- readRDS(file= paste0(output_obj_dir, &quot;340821_Mu_Lung_snRNAseq_merged_annotated.rds&quot; ))</code></pre>
</div>
<div id="color-vectors---define-color-palete" class="section level2">
<h2>Color vectors - Define color palete</h2>
<pre class="r"><code>cols&lt;- pal_igv()(51)
names(cols) &lt;- c(0:50)

colCond &lt;- c(&quot;#631879FF&quot;,&quot;#F16913&quot;)
names(colCond) &lt;- unique(LungMu_obj$condition)


colLab &lt;- rep(&quot;&quot;,length(unique(LungMu_obj$label)))
names(colLab) &lt;- unique(LungMu_obj$label)
colLab[&quot;AT2 Epithelial&quot;] &lt;-&quot;#92C5DE&quot;
colLab[&quot;Bas/SMG duct Epithelial 1&quot;] &lt;-&quot;#F4A582&quot; 
colLab[&quot;Bas/SMG duct Epithelial 2&quot;] &lt;-&quot;#FDDBC7&quot; 
colLab[&quot;AT1 Epithelial&quot;] &lt;-&quot;#9467BDFF&quot;
colLab[&quot;Fibroblasts&quot;] &lt;- &quot;#FF1463FF&quot;
colLab[&quot;Arterial/Cap pulmonary EC&quot;] &lt;-&quot;#990080FF&quot;
colLab[&quot;Alveolar Mph&quot;] &lt;-&quot;#00CC33FF&quot;
colLab[&quot;Cil Epithelial&quot;] &lt;- &quot;#009966FF&quot;
colLab[&quot;Immune cells&quot;] &lt;- &quot;#053061&quot;
colLab[&quot;aCap pulmonary EC&quot;] &lt;- &quot;#0099CCFF&quot;
colLab[&quot;Mesothelial&quot;] &lt;- &quot;#67001F&quot;
colLab[&quot;VSMC/Pericytes&quot;] &lt;- &quot;#E7C76FFF&quot;


colLab_reorder &lt;- colLab[order(factor(names(colLab), levels = c(&quot;Arterial/Cap pulmonary EC&quot;,&quot;aCap pulmonary EC&quot;,
                                                                &quot;Bas/SMG duct Epithelial 1&quot;,&quot;Bas/SMG duct Epithelial 2&quot;,
                                                                &quot;AT1 Epithelial&quot;,&quot;AT2 Epithelial&quot;, &quot;Cil Epithelial&quot;,
                                                                &quot;Fibroblasts&quot;,&quot;VSMC/Pericytes&quot;,&quot;Mesothelial&quot;,
                                                                &quot;Alveolar Mph&quot;, &quot;Immune cells&quot;)))]


LungMu_obj$label &lt;- factor(LungMu_obj$label, levels = names(colLab_reorder))</code></pre>
</div>
<div
id="figure-s2d---compute-differential-gene-expression-average-log2fc-per-cluster-and-per-condition"
class="section level2">
<h2>Figure S2D - Compute differential gene expression (average log2FC
per cluster and per condition)</h2>
<pre class="r"><code>#Add a new column in data
LungMu_obj$celltype_condition &lt;- paste0(LungMu_obj$label,&quot;_&quot;,LungMu_obj$condition)
DefaultAssay(LungMu_obj) &lt;-&quot;RNA&quot;
Idents(LungMu_obj) &lt;- LungMu_obj$celltype_condition
DEmarkers &lt;-FindAllMarkers(LungMu_obj, only.pos=T, logfc.threshold = 0.25,
                           min.pct = 0.25)</code></pre>
<pre class="r"><code>new &lt;- DEmarkers %&gt;%
  separate(cluster,
           into = c(&quot;celltype&quot;,&quot;condition&quot;),
           sep = &quot;_&quot;)
new &lt;-new[new$p_val_adj &lt;= 0.05,] #keep only significant values
#within groups order in descending order
new &lt;- new %&gt;%
  group_by(celltype,condition) %&gt;%
  arrange(desc(avg_log2FC), .by_group = TRUE)

new$condition &lt;- factor(new$condition, levels =c(&quot;Control&quot;, &quot;PH-cHFpEF&quot;))

new$celltype &lt;-factor(new$celltype, levels = names(colLab_reorder))

p2 &lt;- ggplot(new, aes(x=condition,y=avg_log2FC,fill=condition))+
  geom_boxplot(show.legend = TRUE ) + labs(title=&quot;control vs PH-cHFpEF&quot;)  +facet_wrap(~celltype, ncol= 12) + theme(axis.line = element_line(),axis.text.x = element_text(face=&quot;bold&quot;),
    panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(),
    panel.background = element_blank(),
    legend.key=element_blank(),
    axis.text = element_text(angle = 90)) + scale_fill_manual(values = colCond) + xlab(&quot; &quot;) + ylab(&quot;AvgLog2FC&quot;)+
  geom_signif(comparisons = list(c(&quot;Control&quot;,&quot;PH-cHFpEF&quot;)),
              test = &quot;wilcox.test&quot;, step_increase = 0.075, #wilcox
              map_signif_level = TRUE, tip_length = 0, test.args = list(paired = FALSE)) +
  geom_quasirandom(aes(fill = condition), shape = 21, 
                groupOnX = TRUE, size = 0.8, alpha = 0.5) 

g &lt;- grid.force(ggplotGrob(p2))</code></pre>
<p><img src="figure/Alveolar_Mph_analysis.Rmd/boxplot%20with%20average%20log2FC%20and%20significance-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Get the names of grobs and their gPaths into a data.frame structure
grobs_df &lt;- do.call(cbind.data.frame, grid.ls(g, print = FALSE))
# Build optimal gPaths that will be later used to identify grobs and edit them
grobs_df$gPath_full &lt;- paste(grobs_df$gPath, grobs_df$name, sep = &quot;::&quot;)
grobs_df$gPath_full &lt;- gsub(pattern = &quot;layout::&quot;, 
                            replacement = &quot;&quot;, 
                            x = grobs_df$gPath_full, 
                            fixed = TRUE)

strip_bg_gpath &lt;- grobs_df$gPath_full[grepl(pattern = &quot;.*strip\\.background.*&quot;, 
                                            x = grobs_df$gPath_full)]

strip_txt_gpath &lt;- grobs_df$gPath_full[grepl(pattern = &quot;strip.*titleGrob.*text.*&quot;, 
                                             x = grobs_df$gPath_full)]

# Generate some color
n_cols &lt;- length(strip_bg_gpath)
fills &lt;- colLab_reorder #colLab_reorder #rainbow(n_cols)
txt_colors &lt;- gray(0:n_cols/n_cols)

# Edit the grobs
for (i in 1:length(strip_bg_gpath)){
  g &lt;- editGrob(grob = g, gPath = strip_bg_gpath[i], gp = gpar(fill = fills[i]))
  g &lt;- editGrob(grob = g, gPath = strip_txt_gpath[i], gp = gpar(col = txt_colors[i]))
}

# Draw the edited plot
grid.newpage(); grid.draw(g)</code></pre>
<p><img src="figure/Alveolar_Mph_analysis.Rmd/boxplot%20with%20average%20log2FC-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
<div
id="figure-2c---create-an-ma-plot-for-alveolar-macrophages---hfd-vs-cd"
class="section level2">
<h2>Figure 2C - Create an MA plot for alveolar macrophages - HFD vs
CD</h2>
<pre class="r"><code>cl &lt;- &quot;Alveolar Mph&quot;
DefaultAssay(LungMu_obj) &lt;- &quot;RNA&quot;
Idents(LungMu_obj) &lt;- &quot;celltype_condition&quot;
markers &lt;- FindMarkers(LungMu_obj, ident.1 = paste0(cl,&#39;_&#39;,&quot;PH-cHFpEF&quot;), ident.2 = paste0(cl,&#39;_&#39;,&quot;Control&quot;), min.pct = 0.25, logfc.threshold = 0.25)

# Compute mean expression per cell type
markers_df &lt;- markers

# Split gene from ensembl ID
markers_df$gene &lt;- rownames(markers_df)
markers_df &lt;- markers_df  %&gt;% 
    mutate(geneSYMBOL=gsub(&quot;^[^\\.]*\\.&quot;, &quot;&quot;, gene)) #remove all chatacters before first . #old =  &quot;^.*\\.&quot;


# Calculate mean gene/transcript expression only for cells included in each cluster
gene_expression &lt;- GetAssayData(LungMu_obj, layer = &quot;counts&quot;, assay = &quot;RNA&quot;)
geneLogSums &lt;- log2(rowSums(gene_expression))


markers_df$logSum &lt;-geneLogSums[markers_df$gene];
markers_df$diffexpressed &lt;- &#39;NO&#39;
markers_df$diffexpressed[markers_df$avg_log2FC &gt; 0.25 &amp; markers_df$p_val &lt; 0.05 ] &lt;- &#39;UP&#39;
markers_df$diffexpressed[markers_df$avg_log2FC &lt; - 0.25 &amp; markers_df$p_val &lt; 0.05 ] &lt;- &#39;DOWN&#39;

# select to plot all genes in HFD condition and top 30 in control
topdegs_HFD &lt;- markers_df  %&gt;%
  filter(diffexpressed == &quot;UP&quot;) %&gt;%
  select(geneSYMBOL)
topdegs_HFD &lt;- topdegs_HFD[,&quot;geneSYMBOL&quot;][1:50]

topdegs_CD &lt;- markers_df  %&gt;%
  filter(diffexpressed == &quot;DOWN&quot;) %&gt;%
  select(geneSYMBOL)
topdegs_CD &lt;- topdegs_CD[,&quot;geneSYMBOL&quot;][1:40]

topdegs_CD &lt;- topdegs_CD[! topdegs_CD %in% c(&quot;2900064K03Rik&quot;, &quot;9930111J21Rik2&quot;)]

markers_df$delabel &lt;- ifelse((markers_df$diffexpressed %in% c(&quot;UP&quot;, &quot;DOWN&quot;)) &amp;
                                (markers_df$geneSYMBOL %in% topdegs_HFD) |markers_df$geneSYMBOL %in% topdegs_CD,markers_df$geneSYMBOL ,NA)


## Visualize MA plot
p&lt;- ggplot(markers_df,aes(x=logSum, y=avg_log2FC, col=diffexpressed, label = delabel)) +
  # geom_vline(xintercept = c(-2.5, 2.5), col = &#39;gray&#39;, linetype = &#39;dashed&#39;) +
    geom_hline(yintercept = c(0.25), col = &#39;gray&#39;, linetype = &#39;dashed&#39;) +
    geom_point(size=1.5) +
    scale_color_manual(values = c(&quot;#631879FF&quot;, &quot;grey&quot;, &quot;#F16913&quot;),
                       labels = c(&quot;Control&quot;, &quot;Not significant&quot;, &quot;PH-cHFpEF&quot;)
                       ) +
  labs(color = cl ,x = expression(&quot;Mean Normalized Expression&quot;), y = expression(&quot; Average log&quot;[2]*&quot;FC&quot;)) +
  geom_text_repel(max.overlaps = Inf) + theme_set(theme_classic(base_size = 10)) +
  theme(axis.line = element_line(colour = &quot;black&quot;),
    panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(),
    panel.border = element_blank(), 
    panel.background = element_blank()) +
  coord_flip()

print(p)</code></pre>
<p><img src="figure/Alveolar_Mph_analysis.Rmd/alveolar%20Mph%20MA%20HFD%20vs%20cd-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="remove-ensembl-id-from-gene-name" class="section level2">
<h2>Remove ensembl ID from gene name</h2>
<pre class="r"><code>LungMu_obj &lt;- as.SingleCellExperiment(LungMu_obj, assay = &quot;RNA&quot;) #assay = &quot;RNA&quot;

gene_name_fixdf &lt;- data.frame(Gene_Name = rownames(LungMu_obj))
gene_name_fixdf$Symbol &lt;- gsub(&quot;^[^.]*.&quot;,&quot;&quot;, gene_name_fixdf$Gene_Name)

# Check duplicates in df and fix them
duplidx &lt;- rownames((gene_name_fixdf[duplicated(gene_name_fixdf$Symbol), ]))
while (length(duplidx) &gt; 0) {
  Fixed_dupl &lt;- gene_name_fixdf %&gt;% 
  filter(row_number() %in% duplidx)  %&gt;% 
  mutate(Symbol = paste0(Symbol, &quot;.1&quot;))
  gene_name_fixdf[duplidx,&quot;Symbol&quot;] &lt;- Fixed_dupl$Symbol
  
  # Check again for double duplicates
  duplidx &lt;- rownames((gene_name_fixdf[duplicated(gene_name_fixdf$Symbol), ]))
}

rownames(LungMu_obj) &lt;- gene_name_fixdf$Symbol

# Convert SingleCell experiment object back to seurat object by retaining the reductions, normalization, etc
LungMu_obj &lt;- as.Seurat(LungMu_obj, assay = NULL) #assay = NULL</code></pre>
</div>
</div>
<div
id="compute-and-visualize-number-of-alveolar-macrophages-per-condition"
class="section level1">
<h1>Compute and visualize number of Alveolar macrophages per
condition</h1>
<pre class="r"><code>LungMu_obj_Alv_Mph &lt;-subset(LungMu_obj, label == cl) 

cells_per_dataset &lt;- table(LungMu_obj_Alv_Mph$condition)
cells_per_datasetDF &lt;-  data.frame(
  sample=names(cells_per_dataset) ,  
  cells=as.vector(cells_per_dataset)
  )

# Compute percentages
cells_per_datasetDF$fraction &lt;- cells_per_datasetDF$cells / sum(cells_per_datasetDF$cells)

# Compute the cumulative percentages (top of each rectangle)
cells_per_datasetDF$ymax &lt;- cumsum(cells_per_datasetDF$fraction)

# Compute the bottom of each rectangle
cells_per_datasetDF$ymin &lt;- c(0, head(cells_per_datasetDF$ymax, n=-1))

# Compute label position
cells_per_datasetDF$labelPosition &lt;- (cells_per_datasetDF$ymax + cells_per_datasetDF$ymin) / 2

# Compute a good label
cells_per_datasetDF$label &lt;- paste0(cells_per_datasetDF$patient, &quot;\n value: &quot;, cells_per_datasetDF$nuclei)

p&lt;- ggplot(cells_per_datasetDF, aes(ymax=ymax, ymin=ymin, xmax=4, xmin=3, fill=sample)) +
  geom_rect() +
  scale_fill_manual(values = colCond) +
  coord_polar(theta=&quot;y&quot;, start = 0) +
  xlim(c(2, 4)) +
  theme_void() +
  annotate(geom = &#39;text&#39;, x = 2, y = 0, label = as.character(dim(LungMu_obj_Alv_Mph)[2]), size = 10)
p</code></pre>
<p><img src="figure/Alveolar_Mph_analysis.Rmd/Alveolar%20Mph%20abundance-1.png" width="672" style="display: block; margin: auto;" /></p>
<div
id="figure-2d---gene-expression-alterations-among-ph-chfpef-and-control-mice"
class="section level2 tabset">
<h2 class="tabset">Figure 2D - Gene expression alterations among
PH-cHFpEF and control mice</h2>
<div id="cebpb-dusp1-flna-cbr2" class="section level3">
<h3>Cebpb, Dusp1, Flna, Cbr2</h3>
<pre class="r"><code>DefaultAssay(LungMu_obj_Alv_Mph) &lt;- &quot;RNA&quot;

vp_case1 &lt;- function(gene_signature, test_sign, y_max, name, signif){
  plot_case1 &lt;- function(signature){
    VlnPlot(LungMu_obj_Alv_Mph, features = signature,
            pt.size = 0.1, 
           group.by = &quot;condition&quot;, 
            y.max = y_max, # add the y-axis maximum value - otherwise p-value hidden
    cols = colCond) + stat_compare_means(comparisons = test_sign, label = signif, method = &quot;wilcox.test&quot;, paired = FALSE)
    
  }
  purrr::map(gene_signature, plot_case1) %&gt;% cowplot::plot_grid(plotlist = .)
}

comparisons &lt;- list(c(&quot;Control&quot;, &quot;PH-cHFpEF&quot;)) 
  
gene_sig &lt;-c(&quot;Cebpb&quot;, &quot;Dusp1&quot;,&quot;Flna&quot;, &quot;Cbr2&quot; )
vp_case1(gene_signature = gene_sig, test_sign = comparisons, y_max = 7, name = &quot;Violin_plot_AlvMph_significant_1&quot;, signif = &quot;p.signif&quot;)</code></pre>
<p><img src="figure/Alveolar_Mph_analysis.Rmd/violin%20plots%20HFD%20vs%20CD%20-%20Cebpb,%20Dusp1,%20Flna,%20Cbr2-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="sgk1-cd300lf-fos-jund" class="section level3">
<h3>Sgk1, Cd300lf, Fos, Jund</h3>
<pre class="r"><code>gene_sig &lt;-c(&quot;Sgk1&quot;, &quot;Cd300lf&quot;,&quot;Fos&quot;, &quot;Jund&quot; )
vp_case1(gene_signature = gene_sig, test_sign = comparisons, y_max = 7, name = &quot;Violin_plot_AlvMph_significant_2&quot;, signif = &quot;p.signif&quot;)</code></pre>
<p><img src="figure/Alveolar_Mph_analysis.Rmd/violin%20plots%20HFD%20vs%20CD%20-%20Sgk1,%20Cd300lf,%20Fos,%20Jund-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="combined-violin-plot-across-all-clusters"
class="section level2">
<h2>Combined Violin plot across all clusters</h2>
<pre class="r"><code>Idents(LungMu_obj) &lt;- &quot;label&quot;
VlnPlot(object = LungMu_obj, features = c(&quot;Itgax&quot;,&quot;Flna&quot;, &quot;Sgk1&quot;, &quot;Syne2&quot;, &quot;Ccn1&quot;, &quot;Dusp1&quot;, &quot;Fos&quot;, 
                                          &quot;Btg2&quot;, &quot;Actg1&quot;, &quot;Psap&quot;, &quot;Ubc&quot;, &quot;Junb&quot;, &quot;Colgalt1&quot;,
                                          &quot;Axl&quot;,&quot;Cd300lf&quot;, &quot;Amz1&quot;, &quot;Dmxl2&quot;, &quot;Sirpa&quot;, &quot;Cebpb&quot;, &quot;Abcg1&quot;,
                                          &quot;Mertk&quot;, &quot;Chil3&quot;, &quot;Lilr4b&quot;, &quot;F7&quot;, &quot;Ccl6&quot;, &quot;Clec7a&quot;, &quot;Mrc1&quot;, &quot;Csf2ra&quot;, &quot;Tfec&quot;), 
        group.by = &quot;label&quot;, stack = TRUE, split.by = &quot;condition&quot;)+
theme(axis.text.x = element_text(angle = 90)) +
  scale_fill_manual(values = colCond)</code></pre>
<p><img src="figure/Alveolar_Mph_analysis.Rmd/combined%20violin%20plot-1.png" width="1152" style="display: block; margin: auto;" /></p>
</div>
<div id="save-processed-seurat-object" class="section level2">
<h2>Save processed seurat object</h2>
<pre class="r"><code>saveRDS(LungMu_obj_Alv_Mph, file= paste0(output_obj_dir, &quot;340821_Alveolar_macrophages_snRNAseq.rds&quot; ))</code></pre>
</div>
<div id="session-information" class="section level2">
<h2>Session Information</h2>
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.3.1 (2023-06-16)
Platform: aarch64-apple-darwin20 (64-bit)
Running under: macOS 15.3.2

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRblas.0.dylib 
LAPACK: /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.11.0

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

time zone: Europe/Zurich
tzcode source: internal

attached base packages:
[1] grid      stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] ggpubr_0.6.0       ggrepel_0.9.6      ggbeeswarm_0.7.2   ggsignif_0.6.4    
 [5] gridExtra_2.3      scCustomize_1.1.3  viridis_0.6.4      viridisLite_0.4.2 
 [9] pheatmap_1.0.12    ggsci_3.0.0        zeallot_0.1.0      openxlsx_4.2.5.2  
[13] here_1.0.1         lubridate_1.9.4    forcats_1.0.0      purrr_1.0.2       
[17] readr_2.1.4        tidyr_1.3.1        tibble_3.2.1       ggplot2_3.5.2     
[21] tidyverse_2.0.0    dplyr_1.1.4        stringr_1.5.0      SeuratObject_5.0.2
[25] Seurat_4.3.0.1     patchwork_1.3.0    plyr_1.8.9         workflowr_1.7.1   

loaded via a namespace (and not attached):
  [1] RcppAnnoy_0.0.21            splines_4.3.1              
  [3] later_1.3.1                 bitops_1.0-7               
  [5] polyclip_1.10-4             janitor_2.2.0              
  [7] lifecycle_1.0.4             rstatix_0.7.2              
  [9] rprojroot_2.0.3             globals_0.16.3             
 [11] processx_3.8.2              lattice_0.21-8             
 [13] MASS_7.3-60                 backports_1.4.1            
 [15] magrittr_2.0.3              limma_3.56.2               
 [17] plotly_4.10.2               sass_0.4.7                 
 [19] rmarkdown_2.24              jquerylib_0.1.4            
 [21] yaml_2.3.7                  httpuv_1.6.11              
 [23] sctransform_0.3.5           spam_2.10-0                
 [25] zip_2.3.0                   sp_2.1-4                   
 [27] spatstat.sparse_3.0-2       reticulate_1.31            
 [29] cowplot_1.1.1               pbapply_1.7-2              
 [31] RColorBrewer_1.1-3          zlibbioc_1.46.0            
 [33] abind_1.4-5                 GenomicRanges_1.52.0       
 [35] Rtsne_0.16                  RCurl_1.98-1.12            
 [37] BiocGenerics_0.46.0         git2r_0.36.2               
 [39] GenomeInfoDbData_1.2.10     circlize_0.4.15            
 [41] IRanges_2.34.1              S4Vectors_0.38.1           
 [43] irlba_2.3.5.1               listenv_0.9.1              
 [45] spatstat.utils_3.0-5        goftest_1.2-3              
 [47] spatstat.random_3.1-5       fitdistrplus_1.1-11        
 [49] parallelly_1.37.1           DelayedArray_0.26.7        
 [51] leiden_0.4.3                codetools_0.2-19           
 [53] tidyselect_1.2.0            shape_1.4.6                
 [55] farver_2.1.1                stats4_4.3.1               
 [57] matrixStats_1.0.0           spatstat.explore_3.2-1     
 [59] jsonlite_1.8.7              ellipsis_0.3.2             
 [61] progressr_0.14.0            ggridges_0.5.4             
 [63] survival_3.5-5              tools_4.3.1                
 [65] ica_1.0-3                   Rcpp_1.0.13                
 [67] glue_1.8.0                  xfun_0.40                  
 [69] MatrixGenerics_1.12.3       GenomeInfoDb_1.36.2        
 [71] withr_3.0.2                 fastmap_1.1.1              
 [73] fansi_1.0.4                 callr_3.7.3                
 [75] digest_0.6.36               timechange_0.3.0           
 [77] R6_2.5.1                    mime_0.12                  
 [79] ggprism_1.0.4               colorspace_2.1-0           
 [81] scattermore_1.2             tensor_1.5                 
 [83] spatstat.data_3.0-1         utf8_1.2.3                 
 [85] generics_0.1.3              data.table_1.14.8          
 [87] S4Arrays_1.0.6              httr_1.4.6                 
 [89] htmlwidgets_1.6.2           whisker_0.4.1              
 [91] uwot_0.1.16                 pkgconfig_2.0.3            
 [93] gtable_0.3.3                lmtest_0.9-40              
 [95] XVector_0.40.0              SingleCellExperiment_1.22.0
 [97] htmltools_0.5.6             carData_3.0-5              
 [99] dotCall64_1.1-1             Biobase_2.60.0             
[101] scales_1.3.0                png_0.1-8                  
[103] snakecase_0.11.0            knitr_1.43                 
[105] rstudioapi_0.15.0           tzdb_0.4.0                 
[107] reshape2_1.4.4              nlme_3.1-162               
[109] cachem_1.0.8                zoo_1.8-12                 
[111] GlobalOptions_0.1.2         KernSmooth_2.23-21         
[113] parallel_4.3.1              miniUI_0.1.1.1             
[115] vipor_0.4.5                 ggrastr_1.0.2              
[117] pillar_1.9.0                vctrs_0.6.5                
[119] RANN_2.6.1                  promises_1.2.1             
[121] car_3.1-2                   xtable_1.8-4               
[123] cluster_2.1.4               beeswarm_0.4.0             
[125] paletteer_1.6.0             evaluate_0.21              
[127] cli_3.6.3                   compiler_4.3.1             
[129] crayon_1.5.3                rlang_1.1.4                
[131] future.apply_1.11.2         labeling_0.4.2             
[133] rematch2_2.1.2              ps_1.7.5                   
[135] getPass_0.2-4               fs_1.6.3                   
[137] stringi_1.7.12              deldir_1.0-9               
[139] munsell_0.5.0               lazyeval_0.2.2             
[141] spatstat.geom_3.2-4         Matrix_1.6-4               
[143] hms_1.1.3                   future_1.33.2              
[145] shiny_1.7.5                 SummarizedExperiment_1.30.2
[147] highr_0.10                  ROCR_1.0-11                
[149] igraph_1.5.1                broom_1.0.5                
[151] bslib_0.5.1                </code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.3.1 (2023-06-16)
Platform: aarch64-apple-darwin20 (64-bit)
Running under: macOS 15.3.2

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRblas.0.dylib 
LAPACK: /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.11.0

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

time zone: Europe/Zurich
tzcode source: internal

attached base packages:
[1] grid      stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] ggpubr_0.6.0       ggrepel_0.9.6      ggbeeswarm_0.7.2   ggsignif_0.6.4    
 [5] gridExtra_2.3      scCustomize_1.1.3  viridis_0.6.4      viridisLite_0.4.2 
 [9] pheatmap_1.0.12    ggsci_3.0.0        zeallot_0.1.0      openxlsx_4.2.5.2  
[13] here_1.0.1         lubridate_1.9.4    forcats_1.0.0      purrr_1.0.2       
[17] readr_2.1.4        tidyr_1.3.1        tibble_3.2.1       ggplot2_3.5.2     
[21] tidyverse_2.0.0    dplyr_1.1.4        stringr_1.5.0      SeuratObject_5.0.2
[25] Seurat_4.3.0.1     patchwork_1.3.0    plyr_1.8.9         workflowr_1.7.1   

loaded via a namespace (and not attached):
  [1] RcppAnnoy_0.0.21            splines_4.3.1              
  [3] later_1.3.1                 bitops_1.0-7               
  [5] polyclip_1.10-4             janitor_2.2.0              
  [7] lifecycle_1.0.4             rstatix_0.7.2              
  [9] rprojroot_2.0.3             globals_0.16.3             
 [11] processx_3.8.2              lattice_0.21-8             
 [13] MASS_7.3-60                 backports_1.4.1            
 [15] magrittr_2.0.3              limma_3.56.2               
 [17] plotly_4.10.2               sass_0.4.7                 
 [19] rmarkdown_2.24              jquerylib_0.1.4            
 [21] yaml_2.3.7                  httpuv_1.6.11              
 [23] sctransform_0.3.5           spam_2.10-0                
 [25] zip_2.3.0                   sp_2.1-4                   
 [27] spatstat.sparse_3.0-2       reticulate_1.31            
 [29] cowplot_1.1.1               pbapply_1.7-2              
 [31] RColorBrewer_1.1-3          zlibbioc_1.46.0            
 [33] abind_1.4-5                 GenomicRanges_1.52.0       
 [35] Rtsne_0.16                  RCurl_1.98-1.12            
 [37] BiocGenerics_0.46.0         git2r_0.36.2               
 [39] GenomeInfoDbData_1.2.10     circlize_0.4.15            
 [41] IRanges_2.34.1              S4Vectors_0.38.1           
 [43] irlba_2.3.5.1               listenv_0.9.1              
 [45] spatstat.utils_3.0-5        goftest_1.2-3              
 [47] spatstat.random_3.1-5       fitdistrplus_1.1-11        
 [49] parallelly_1.37.1           DelayedArray_0.26.7        
 [51] leiden_0.4.3                codetools_0.2-19           
 [53] tidyselect_1.2.0            shape_1.4.6                
 [55] farver_2.1.1                stats4_4.3.1               
 [57] matrixStats_1.0.0           spatstat.explore_3.2-1     
 [59] jsonlite_1.8.7              ellipsis_0.3.2             
 [61] progressr_0.14.0            ggridges_0.5.4             
 [63] survival_3.5-5              tools_4.3.1                
 [65] ica_1.0-3                   Rcpp_1.0.13                
 [67] glue_1.8.0                  xfun_0.40                  
 [69] MatrixGenerics_1.12.3       GenomeInfoDb_1.36.2        
 [71] withr_3.0.2                 fastmap_1.1.1              
 [73] fansi_1.0.4                 callr_3.7.3                
 [75] digest_0.6.36               timechange_0.3.0           
 [77] R6_2.5.1                    mime_0.12                  
 [79] ggprism_1.0.4               colorspace_2.1-0           
 [81] scattermore_1.2             tensor_1.5                 
 [83] spatstat.data_3.0-1         utf8_1.2.3                 
 [85] generics_0.1.3              data.table_1.14.8          
 [87] S4Arrays_1.0.6              httr_1.4.6                 
 [89] htmlwidgets_1.6.2           whisker_0.4.1              
 [91] uwot_0.1.16                 pkgconfig_2.0.3            
 [93] gtable_0.3.3                lmtest_0.9-40              
 [95] XVector_0.40.0              SingleCellExperiment_1.22.0
 [97] htmltools_0.5.6             carData_3.0-5              
 [99] dotCall64_1.1-1             Biobase_2.60.0             
[101] scales_1.3.0                png_0.1-8                  
[103] snakecase_0.11.0            knitr_1.43                 
[105] rstudioapi_0.15.0           tzdb_0.4.0                 
[107] reshape2_1.4.4              nlme_3.1-162               
[109] cachem_1.0.8                zoo_1.8-12                 
[111] GlobalOptions_0.1.2         KernSmooth_2.23-21         
[113] parallel_4.3.1              miniUI_0.1.1.1             
[115] vipor_0.4.5                 ggrastr_1.0.2              
[117] pillar_1.9.0                vctrs_0.6.5                
[119] RANN_2.6.1                  promises_1.2.1             
[121] car_3.1-2                   xtable_1.8-4               
[123] cluster_2.1.4               beeswarm_0.4.0             
[125] paletteer_1.6.0             evaluate_0.21              
[127] cli_3.6.3                   compiler_4.3.1             
[129] crayon_1.5.3                rlang_1.1.4                
[131] future.apply_1.11.2         labeling_0.4.2             
[133] rematch2_2.1.2              ps_1.7.5                   
[135] getPass_0.2-4               fs_1.6.3                   
[137] stringi_1.7.12              deldir_1.0-9               
[139] munsell_0.5.0               lazyeval_0.2.2             
[141] spatstat.geom_3.2-4         Matrix_1.6-4               
[143] hms_1.1.3                   future_1.33.2              
[145] shiny_1.7.5                 SummarizedExperiment_1.30.2
[147] highr_0.10                  ROCR_1.0-11                
[149] igraph_1.5.1                broom_1.0.5                
[151] bslib_0.5.1                </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
