<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Iliana Papadopoulou" />

<meta name="date" content="2025-05-11" />

<title>Murine Lung Landscape - Preprocessing</title>

<script src="site_libs/header-attrs-2.24/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Alveolar_Mph_mouse_lung_PH</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Murine Lung Landscape - Preprocessing</h1>
<h4 class="author">Iliana Papadopoulou</h4>
<h4 class="date">2025-05-11</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2025-05-11
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong>
<code>Alveolar_Mph_mouse_lung_PH/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20250507code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20250507)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20250507code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20250507)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomilianapapAlveolarMphmouselungPHtree2b8e55a23162b6873ed17cba304c7564bef86591targetblank2b8e55aa">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/iliana-pap/Alveolar_Mph_mouse_lung_PH/tree/2b8e55a23162b6873ed17cba304c7564bef86591" target="_blank">2b8e55a</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomilianapapAlveolarMphmouselungPHtree2b8e55a23162b6873ed17cba304c7564bef86591targetblank2b8e55aa"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/iliana-pap/Alveolar_Mph_mouse_lung_PH/tree/2b8e55a23162b6873ed17cba304c7564bef86591" target="_blank">2b8e55a</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    data/Output/

Unstaged changes:
    Modified:   analysis/Alveolar_Macrophages_GSEA.Rmd
    Modified:   analysis/Alveolar_Mph_analysis.Rmd
    Modified:   analysis/MuLung_snRNAseq_analysis.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown
(<code>analysis/MuLung_snRNAseq_preprocessing.Rmd</code>) and HTML
(<code>docs/MuLung_snRNAseq_preprocessing.html</code>) files. If you’ve
configured a remote Git repository (see <code>?wflow_git_remote</code>),
click on the hyperlinks in the table below to view the files as they
were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/iliana-pap/Alveolar_Mph_mouse_lung_PH/blob/2b8e55a23162b6873ed17cba304c7564bef86591/analysis/MuLung_snRNAseq_preprocessing.Rmd" target="_blank">2b8e55a</a>
</td>
<td>
iliana-pap
</td>
<td>
2025-05-11
</td>
<td>
update preprocessing
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/iliana-pap/Alveolar_Mph_mouse_lung_PH/6933769e8e13c56ea87bb35c20b0d3aead2afbf8/docs/MuLung_snRNAseq_preprocessing.html" target="_blank">6933769</a>
</td>
<td>
iliana-pap
</td>
<td>
2025-05-07
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/iliana-pap/Alveolar_Mph_mouse_lung_PH/blob/32d33de80fca5eb19af542af6f4709ac8187f372/analysis/MuLung_snRNAseq_preprocessing.Rmd" target="_blank">32d33de</a>
</td>
<td>
iliana-pap
</td>
<td>
2025-05-07
</td>
<td>
Perform quality control on murine samples
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/iliana-pap/Alveolar_Mph_mouse_lung_PH/257493bf49f1369fc7fc269f0df51991f22691c1/docs/MuLung_snRNAseq_preprocessing.html" target="_blank">257493b</a>
</td>
<td>
iliana-pap
</td>
<td>
2025-05-07
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/iliana-pap/Alveolar_Mph_mouse_lung_PH/blob/5e281fe63ef1b6106ccb7adb2e8664d0dee12f24/analysis/MuLung_snRNAseq_preprocessing.Rmd" target="_blank">5e281fe</a>
</td>
<td>
iliana-pap
</td>
<td>
2025-05-07
</td>
<td>
Create project
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div
id="preprocessing-of-murine-lung-samples-from-normal-diet-vs-mice-with-high-fat-diet-l-name-treatment-with-hfpef"
class="section level1">
<h1>1. <strong>Preprocessing of murine lung samples from normal diet vs
mice with high fat diet + L-name treatment with HFpEF</strong></h1>
<div id="load-r-libraries" class="section level2">
<h2>Load R libraries</h2>
<pre class="r"><code>suppressPackageStartupMessages({
  library(plyr)
  library(patchwork)
  library(Seurat)
  library(stringr)
  library(dplyr)
  library(tidyverse)
  library(here)
  library(openxlsx)
  library(zeallot)
  library(ggsci)
  library(openxlsx)
  library(pheatmap)
  library(stringr)
  library(viridis)
  library(scCustomize)
  library(stringr)
  library(gridExtra)
  library(grid)
})</code></pre>
</div>
<div id="set-directory" class="section level2">
<h2>Set directory</h2>
<pre class="r"><code>basedir &lt;- here()
source(paste0(basedir,&#39;/analysis/Lung_snRNAseq_functions.R&#39;), echo=FALSE)

# Create directory to save preprocessed seurat objects
output_obj_dir &lt;- paste0(basedir,&quot;/data/Output/&quot;)
dir.create(output_obj_dir)</code></pre>
</div>
<div id="read-lung-tissue-samples" class="section level2">
<h2>Read lung tissue samples</h2>
<pre class="r"><code>pattern &lt;- &quot;.*_seurat.*&quot;
files &lt;- list.files(paste0(basedir,&quot;/data&quot;, &quot;/&quot;), pattern = pattern)
LungMu_obj.list &lt;- c()
for (f in files){
  print(f)
  LungMu_obj &lt;- readRDS(file = paste0(basedir,&quot;/data/&quot;,f))
  LungMu_obj.list &lt;- c(LungMu_obj.list,LungMu_obj)
}</code></pre>
<pre><code>[1] &quot;340821_14-1_20240119_Mu_nucseq_lung_CD_1_seurat.rds&quot;
[1] &quot;340821_15-2_20240119_Mu_nucseq_lung_CD_2_seurat.rds&quot;
[1] &quot;340821_16-3_20240119_Mu_nucseq_lung_CD_3_seurat.rds&quot;
[1] &quot;340821_17-4_20240119_Mu_nucseq_lung_HFD_1_seurat.rds&quot;
[1] &quot;340821_18-5_20240119_Mu_nucseq_lung_HFD_2_seurat.rds&quot;
[1] &quot;340821_19-6_20240119_Mu_nucseq_lung_HFD_3_seurat.rds&quot;
[1] &quot;340821_20-7_20240119_Mu_nucseq_lung_HFD_4_seurat.rds&quot;</code></pre>
</div>
<div id="merged-samples-into-one-seurat-object" class="section level2">
<h2>Merged samples into one seurat object</h2>
<pre class="r"><code>LungMu_obj &lt;- merge(LungMu_obj.list[[1]], y = LungMu_obj.list[2:length(LungMu_obj.list)])
print(dim(LungMu_obj))</code></pre>
<pre><code>[1] 26178 17061</code></pre>
</div>
<div id="add-condition-column-in-the-object" class="section level2">
<h2>Add condition column in the object</h2>
<pre class="r"><code># Extract condition
temp &lt;- LungMu_obj@meta.data %&gt;% mutate(condition = case_when(
    str_split_fixed(LungMu_obj@meta.data$dataset, &quot;_&quot;, 8)[,8] == &quot;&quot;  ~ str_split_fixed(LungMu_obj@meta.data$dataset, &quot;_&quot;, 8)[,6] ,
    str_split_fixed(LungMu_obj@meta.data$dataset, &quot;_&quot;, 8)[,8] != &quot;&quot; ~ str_split_fixed(LungMu_obj@meta.data$dataset, &quot;_&quot;, 8)[,7]
  )
)

LungMu_obj@meta.data$condition &lt;- temp$condition</code></pre>
</div>
<div id="alter-condition-names-inside-the-seurat-object"
class="section level2">
<h2>Alter condition names inside the seurat object</h2>
<pre class="r"><code>LungMu_obj$condition[LungMu_obj$condition==&quot;CD&quot;] &lt;- &quot;Control&quot;
LungMu_obj$condition[LungMu_obj$condition==&quot;HFD&quot;] &lt;- &quot;PH-cHFpEF&quot;</code></pre>
</div>
<div id="color-vectors---define-color-palete" class="section level2">
<h2>Color vectors - Define color palete</h2>
<pre class="r"><code>cols&lt;- pal_igv()(51)
names(cols) &lt;- c(0:50)

colDataset &lt;- c(&quot;#FF7F0EFF&quot;,&quot;#17BECFFF&quot;, &quot;#BB0021FF&quot;, &quot;#00CC33FF&quot;, &quot;#3B4992FF&quot;, &quot;#4775FFFF&quot;, &quot;#990080FF&quot;, &quot;#AEC7E8FF&quot;,&quot;#FFBB78FF&quot;,&quot;#1B1919FF&quot;)
names(colDataset) &lt;- unique(LungMu_obj$dataset)

colCond &lt;- c(&quot;#631879FF&quot;,&quot;#F16913&quot;)
names(colCond) &lt;- unique(LungMu_obj$condition)</code></pre>
</div>
<div id="investigate-quality-control-qc-metrics" class="section level2">
<h2>Investigate Quality Control (QC) metrics</h2>
<div id="check-nuclei-and-gene-counts-for-all-samples"
class="section level3 tabset">
<h3 class="tabset">Check nuclei and gene counts for all samples</h3>
<div id="nuclei-counts-per-condition" class="section level4">
<h4>Nuclei counts per condition</h4>
<pre class="r"><code># Visualize the number of nuclei counts per condition
LungMu_obj@meta.data %&gt;% 
    ggplot(aes(x=condition, fill=condition)) + 
    geom_bar() +
    scale_fill_manual(values=colCond) +
    theme_classic() +
    theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust=1)) +
    theme(plot.title = element_text(hjust=0.5, face=&quot;bold&quot;))</code></pre>
<p><img src="figure/MuLung_snRNAseq_preprocessing.Rmd/Nuclei%20counts%20per%20condition-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Nuclei-counts-per-condition-1">
Past versions of "Nuclei counts per condition-1.png"
</button>
</p>
<div id="fig-no-spaces-Nuclei-counts-per-condition-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/iliana-pap/Alveolar_Mph_mouse_lung_PH/blob/6933769e8e13c56ea87bb35c20b0d3aead2afbf8/docs/figure/MuLung_snRNAseq_preprocessing.Rmd/Nuclei counts per condition-1.png" target="_blank">6933769</a>
</td>
<td>
iliana-pap
</td>
<td>
2025-05-07
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="maximum-gene-counts" class="section level4">
<h4>Maximum gene counts</h4>
<pre class="r"><code>mouse_data &lt;- LungMu_obj@meta.data

gene_counts &lt;-as.data.frame(mouse_data %&gt;% group_by(dataset) %&gt;% summarise(max_gene_counts = max(nFeature_RNA)))

p&lt;- ggplot(gene_counts, aes(x=dataset, y=max_gene_counts, fill = dataset)) + 
  geom_bar(stat = &quot;identity&quot;)+
  scale_fill_manual(values = colDataset) +
  theme(axis.line = element_line(colour = &quot;black&quot;),
    panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(),
    panel.border = element_blank(),
    panel.background = element_blank(),
    axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))
print(p)</code></pre>
<p><img src="figure/MuLung_snRNAseq_preprocessing.Rmd/maximum%20gene%20counts-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-maximum-gene-counts-1">
Past versions of "maximum gene counts-1.png"
</button>
</p>
<div id="fig-no-spaces-maximum-gene-counts-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/iliana-pap/Alveolar_Mph_mouse_lung_PH/blob/6933769e8e13c56ea87bb35c20b0d3aead2afbf8/docs/figure/MuLung_snRNAseq_preprocessing.Rmd/maximum gene counts-1.png" target="_blank">6933769</a>
</td>
<td>
iliana-pap
</td>
<td>
2025-05-07
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># Convert to named vector
geneCountsPerDataset &lt;- as.character(gene_counts$max_gene_counts)
names(geneCountsPerDataset) &lt;- gene_counts$dataset
print(geneCountsPerDataset)</code></pre>
<pre><code> 340821_14-1_20240119_Mu_nucseq_lung_CD_1 
                                   &quot;1534&quot; 
 340821_15-2_20240119_Mu_nucseq_lung_CD_2 
                                   &quot;3083&quot; 
 340821_16-3_20240119_Mu_nucseq_lung_CD_3 
                                   &quot;3952&quot; 
340821_17-4_20240119_Mu_nucseq_lung_HFD_1 
                                    &quot;119&quot; 
340821_18-5_20240119_Mu_nucseq_lung_HFD_2 
                                   &quot;3413&quot; 
340821_19-6_20240119_Mu_nucseq_lung_HFD_3 
                                   &quot;2891&quot; 
340821_20-7_20240119_Mu_nucseq_lung_HFD_4 
                                   &quot;1640&quot; </code></pre>
</div>
<div id="nuclei-counts-per-dataset" class="section level4">
<h4>Nuclei counts per dataset</h4>
<pre class="r"><code># Visualize the number of cell counts per condition
LungMu_obj@meta.data %&gt;% 
    ggplot(aes(x=dataset, fill=dataset)) + 
    geom_bar() +
    scale_fill_manual(values=colDataset) +
    theme_classic() +
    theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust=1)) +
    theme(plot.title = element_text(hjust=0.5, face=&quot;bold&quot;))</code></pre>
<p><img src="figure/MuLung_snRNAseq_preprocessing.Rmd/Nuclei%20counts%20per%20dataset-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Nuclei-counts-per-dataset-1">
Past versions of "Nuclei counts per dataset-1.png"
</button>
</p>
<div id="fig-no-spaces-Nuclei-counts-per-dataset-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/iliana-pap/Alveolar_Mph_mouse_lung_PH/blob/6933769e8e13c56ea87bb35c20b0d3aead2afbf8/docs/figure/MuLung_snRNAseq_preprocessing.Rmd/Nuclei counts per dataset-1.png" target="_blank">6933769</a>
</td>
<td>
iliana-pap
</td>
<td>
2025-05-07
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
</div>
<div id="identification-of-highly-variable-features"
class="section level2">
<h2>Identification of highly variable features</h2>
<pre class="r"><code>LungMu_obj &lt;- FindVariableFeatures(LungMu_obj, selection.method = &quot;vst&quot;, nfeatures = 2000)</code></pre>
</div>
<div id="nuclei-and-gene-expression-distributions"
class="section level2 tabset">
<h2 class="tabset">Nuclei and gene expression distributions</h2>
<div id="umis---transcripts-per-dataset" class="section level3">
<h3>UMIs - transcripts per dataset</h3>
<pre class="r"><code>LungMu_obj@meta.data %&gt;% 
    ggplot(aes(color=dataset, x=nCount_RNA, fill= dataset)) +
    scale_fill_manual(values=colDataset) +
    geom_density(alpha = 0.2) + 
    scale_x_log10() + 
    theme_classic() +
    ylab(&quot;Cell density&quot;) +
    geom_vline(xintercept = 300)</code></pre>
<p><img src="figure/MuLung_snRNAseq_preprocessing.Rmd/UMIs%20-%20transcripts%20per%20dataset-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-UMIs---transcripts-per-dataset-1">
Past versions of "UMIs - transcripts per dataset-1.png"
</button>
</p>
<div id="fig-no-spaces-UMIs---transcripts-per-dataset-1"
class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/iliana-pap/Alveolar_Mph_mouse_lung_PH/blob/6933769e8e13c56ea87bb35c20b0d3aead2afbf8/docs/figure/MuLung_snRNAseq_preprocessing.Rmd/UMIs - transcripts per dataset-1.png" target="_blank">6933769</a>
</td>
<td>
iliana-pap
</td>
<td>
2025-05-07
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="gene-distribution-per-nuclei" class="section level3">
<h3>Gene distribution per nuclei</h3>
<pre class="r"><code># Visualize the distribution of genes detected per nuclei via histogram
LungMu_obj@meta.data %&gt;% 
    ggplot(aes(color=dataset, x=nFeature_RNA, fill= dataset)) + 
  scale_fill_manual(values=colDataset) +
    geom_density(alpha = 0.2) + 
        scale_x_log10() +
    theme_classic() +
    geom_vline(xintercept = 150)</code></pre>
<p><img src="figure/MuLung_snRNAseq_preprocessing.Rmd/Gene%20distribution%20per%20nuclei-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Gene-distribution-per-nuclei-1">
Past versions of "Gene distribution per nuclei-1.png"
</button>
</p>
<div id="fig-no-spaces-Gene-distribution-per-nuclei-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/iliana-pap/Alveolar_Mph_mouse_lung_PH/blob/6933769e8e13c56ea87bb35c20b0d3aead2afbf8/docs/figure/MuLung_snRNAseq_preprocessing.Rmd/Gene distribution per nuclei-1.png" target="_blank">6933769</a>
</td>
<td>
iliana-pap
</td>
<td>
2025-05-07
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="genes-detected-per-umi" class="section level3">
<h3>Genes detected per UMI</h3>
<pre class="r"><code># Add number of genes per UMI for each nuclei to metadata - measures complexity of dataset 
LungMu_obj$log10GenesPerUMI &lt;- log10(LungMu_obj$nFeature_RNA) / log10(LungMu_obj$nCount_RNA)

# Visualize the overall complexity of the gene expression by visualizing the genes detected per UMI (novelty score)
LungMu_obj@meta.data %&gt;%
    ggplot(aes(x=log10GenesPerUMI, color = dataset, fill=dataset)) +
    scale_fill_manual(values=colDataset) +
    geom_density(alpha = 0.2) +
    theme_classic() +
    geom_vline(xintercept = 0.88)</code></pre>
<p><img src="figure/MuLung_snRNAseq_preprocessing.Rmd/genes%20detected%20per%20UMI-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-genes-detected-per-UMI-1">
Past versions of "genes detected per UMI-1.png"
</button>
</p>
<div id="fig-no-spaces-genes-detected-per-UMI-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/iliana-pap/Alveolar_Mph_mouse_lung_PH/blob/6933769e8e13c56ea87bb35c20b0d3aead2afbf8/docs/figure/MuLung_snRNAseq_preprocessing.Rmd/genes detected per UMI-1.png" target="_blank">6933769</a>
</td>
<td>
iliana-pap
</td>
<td>
2025-05-07
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div
id="compute-and-visualize-number-of-nuclei-per-sample-via-a-pie-chart"
class="section level2">
<h2>Compute and visualize number of nuclei per sample via a pie
chart</h2>
<pre class="r"><code>cells_per_dataset &lt;- table(LungMu_obj$dataset)
cells_per_datasetDF &lt;-  data.frame(
  sample=names(cells_per_dataset) ,  
  cells=as.vector(cells_per_dataset)
  )

# Compute percentages
cells_per_datasetDF$fraction &lt;- cells_per_datasetDF$cells / sum(cells_per_datasetDF$cells)

# Compute the cumulative percentages (top of each rectangle)
cells_per_datasetDF$ymax &lt;- cumsum(cells_per_datasetDF$fraction)

# Compute the bottom of each rectangle
cells_per_datasetDF$ymin &lt;- c(0, head(cells_per_datasetDF$ymax, n=-1))

# Compute label position
cells_per_datasetDF$labelPosition &lt;- (cells_per_datasetDF$ymax + cells_per_datasetDF$ymin) / 2

# Compute a good label
cells_per_datasetDF$label &lt;- paste0(cells_per_datasetDF$patient, &quot;\n value: &quot;, cells_per_datasetDF$nuclei)

# Make the plot
p&lt;- ggplot(cells_per_datasetDF, aes(ymax=ymax, ymin=ymin, xmax=4, xmin=3, fill=sample)) +
  geom_rect() +
  scale_fill_manual(values = colDataset) +
  coord_polar(theta=&quot;y&quot;, start = 0) +
  xlim(c(2, 4)) +
  theme_void() +
  annotate(geom = &#39;text&#39;, x = 2, y = 0, label = as.character(dim(LungMu_obj)[2]), size = 10)
p</code></pre>
<p><img src="figure/MuLung_snRNAseq_preprocessing.Rmd/number%20of%20nuclei/sample-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-sample-1">
Past versions of sample-1.png
</button>
</p>
<div id="fig-sample-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/iliana-pap/Alveolar_Mph_mouse_lung_PH/blob/6933769e8e13c56ea87bb35c20b0d3aead2afbf8/docs/figure/MuLung_snRNAseq_preprocessing.Rmd/number of nuclei/sample-1.png" target="_blank">6933769</a>
</td>
<td>
iliana-pap
</td>
<td>
2025-05-07
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div
id="find-median-number-of-nuclei-and-genes-per-nuclei-before-filtering"
class="section level2">
<h2>Find median number of nuclei and genes per nuclei before
filtering</h2>
<pre class="r"><code>median_nuclei_per_sample &lt;- LungMu_obj@meta.data %&gt;% group_by(dataset) %&gt;% summarise(across(nCount_RNA, median))
median_nuclei_per_sample</code></pre>
<pre><code># A tibble: 7 × 2
  dataset                                   nCount_RNA
  &lt;chr&gt;                                          &lt;dbl&gt;
1 340821_14-1_20240119_Mu_nucseq_lung_CD_1         521
2 340821_15-2_20240119_Mu_nucseq_lung_CD_2        1152
3 340821_16-3_20240119_Mu_nucseq_lung_CD_3        2513
4 340821_17-4_20240119_Mu_nucseq_lung_HFD_1         81
5 340821_18-5_20240119_Mu_nucseq_lung_HFD_2       1426
6 340821_19-6_20240119_Mu_nucseq_lung_HFD_3       1157
7 340821_20-7_20240119_Mu_nucseq_lung_HFD_4        479</code></pre>
<pre class="r"><code>median_genes_per_sample &lt;- LungMu_obj@meta.data %&gt;% group_by(dataset) %&gt;% summarise(across(nFeature_RNA, median))
median_genes_per_sample</code></pre>
<pre><code># A tibble: 7 × 2
  dataset                                   nFeature_RNA
  &lt;chr&gt;                                            &lt;dbl&gt;
1 340821_14-1_20240119_Mu_nucseq_lung_CD_1          328.
2 340821_15-2_20240119_Mu_nucseq_lung_CD_2          680 
3 340821_16-3_20240119_Mu_nucseq_lung_CD_3         1354 
4 340821_17-4_20240119_Mu_nucseq_lung_HFD_1          37 
5 340821_18-5_20240119_Mu_nucseq_lung_HFD_2         795 
6 340821_19-6_20240119_Mu_nucseq_lung_HFD_3         707 
7 340821_20-7_20240119_Mu_nucseq_lung_HFD_4         290 </code></pre>
</div>
<div id="nuclei-and-gene-distributions-per-condition"
class="section level2 tabset">
<h2 class="tabset">Nuclei and gene distributions per condition</h2>
<div id="umis---transcripts-per-condition" class="section level3">
<h3>UMIs - transcripts per condition</h3>
<pre class="r"><code>LungMu_obj@meta.data %&gt;% 
    ggplot(aes(color=condition, x=nCount_RNA, fill= condition)) +
    scale_fill_manual(values=colCond) +
    geom_density(alpha = 0.2) + 
    scale_x_log10() + 
    theme_classic() +
    ylab(&quot;Cell density&quot;) +
    geom_vline(xintercept = 200)</code></pre>
<p><img src="figure/MuLung_snRNAseq_preprocessing.Rmd/UMIs%20-%20transcripts%20per%20condition-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-UMIs---transcripts-per-condition-1">
Past versions of "UMIs - transcripts per condition-1.png"
</button>
</p>
<div id="fig-no-spaces-UMIs---transcripts-per-condition-1"
class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/iliana-pap/Alveolar_Mph_mouse_lung_PH/blob/6933769e8e13c56ea87bb35c20b0d3aead2afbf8/docs/figure/MuLung_snRNAseq_preprocessing.Rmd/UMIs - transcripts per condition-1.png" target="_blank">6933769</a>
</td>
<td>
iliana-pap
</td>
<td>
2025-05-07
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="gene-distribution-per-condition" class="section level3">
<h3>Gene distribution per condition</h3>
<pre class="r"><code># Visualize the distribution of genes detected per cell via histogram
LungMu_obj@meta.data %&gt;% 
    ggplot(aes(color=condition, x=nFeature_RNA, fill= condition)) + 
  scale_fill_manual(values=colCond) +
    geom_density(alpha = 0.2) + 
    theme_classic() +
    scale_x_log10() + 
    geom_vline(xintercept = 110)</code></pre>
<p><img src="figure/MuLung_snRNAseq_preprocessing.Rmd/Gene%20distribution%20per%20condition-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Gene-distribution-per-condition-1">
Past versions of "Gene distribution per condition-1.png"
</button>
</p>
<div id="fig-no-spaces-Gene-distribution-per-condition-1"
class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/iliana-pap/Alveolar_Mph_mouse_lung_PH/blob/6933769e8e13c56ea87bb35c20b0d3aead2afbf8/docs/figure/MuLung_snRNAseq_preprocessing.Rmd/Gene distribution per condition-1.png" target="_blank">6933769</a>
</td>
<td>
iliana-pap
</td>
<td>
2025-05-07
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="novelty-score" class="section level3">
<h3>novelty score</h3>
<pre class="r"><code># Visualize the overall complexity of the gene expression by visualizing the genes detected per UMI (novelty score)
LungMu_obj@meta.data %&gt;%
    ggplot(aes(x=log10GenesPerUMI, color = condition, fill=condition)) +
    scale_fill_manual(values=colCond) +
    geom_density(alpha = 0.2) +
    theme_classic() +
    geom_vline(xintercept = 0.86)</code></pre>
<p><img src="figure/MuLung_snRNAseq_preprocessing.Rmd/novelty%20score-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-novelty-score-1">
Past versions of "novelty score-1.png"
</button>
</p>
<div id="fig-no-spaces-novelty-score-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/iliana-pap/Alveolar_Mph_mouse_lung_PH/blob/6933769e8e13c56ea87bb35c20b0d3aead2afbf8/docs/figure/MuLung_snRNAseq_preprocessing.Rmd/novelty score-1.png" target="_blank">6933769</a>
</td>
<td>
iliana-pap
</td>
<td>
2025-05-07
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div
id="exclude-hfd-1-sample-from-the-analysis-as-it-has-an-extremely-low-median-of-nuclei-and-genes-per-nuclei"
class="section level2">
<h2>Exclude HFD-1 sample from the analysis as it has an extremely low
median of nuclei and genes per nuclei</h2>
<pre class="r"><code>LungMu_obj &lt;- subset(LungMu_obj, subset = (dataset == &#39;340821_17-4_20240119_Mu_nucseq_lung_HFD_1&#39;), invert = TRUE)</code></pre>
</div>
<div id="plot-qc-metrics-again" class="section level2 tabset">
<h2 class="tabset">Plot QC metrics again</h2>
<div id="umis---transcripts-per-nuclei" class="section level3">
<h3>UMIs - transcripts per nuclei</h3>
<pre class="r"><code>LungMu_obj@meta.data %&gt;% 
    ggplot(aes(color=dataset, x=nCount_RNA, fill= dataset)) +
    scale_fill_manual(values=colDataset) +
    geom_density(alpha = 0.2) + 
    scale_x_log10() + 
    theme_classic() +
    ylab(&quot;Cell density&quot;)</code></pre>
<p><img src="figure/MuLung_snRNAseq_preprocessing.Rmd/UMIs%20-%20transcripts%20per%20nuclei-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-UMIs---transcripts-per-nuclei-1">
Past versions of "UMIs - transcripts per nuclei-1.png"
</button>
</p>
<div id="fig-no-spaces-UMIs---transcripts-per-nuclei-1"
class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/iliana-pap/Alveolar_Mph_mouse_lung_PH/blob/6933769e8e13c56ea87bb35c20b0d3aead2afbf8/docs/figure/MuLung_snRNAseq_preprocessing.Rmd/UMIs - transcripts per nuclei-1.png" target="_blank">6933769</a>
</td>
<td>
iliana-pap
</td>
<td>
2025-05-07
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="gene-distribution-per-nuclei-1" class="section level3">
<h3>Gene distribution per nuclei</h3>
<pre class="r"><code># Visualize the distribution of genes detected per cell via histogram
LungMu_obj@meta.data %&gt;% 
    ggplot(aes(color=dataset, x=nFeature_RNA, fill= dataset)) + 
  scale_fill_manual(values=colDataset) +
    geom_density(alpha = 0.2) + 
        scale_x_log10() +
    theme_classic() </code></pre>
<p><img src="figure/MuLung_snRNAseq_preprocessing.Rmd/CLEANED%20:%20Gene%20distribution%20per%20nuclei-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-CLEANED-:-Gene-distribution-per-nuclei-1">
Past versions of "CLEANED : Gene distribution per nuclei-1.png"
</button>
</p>
<div id="fig-no-spaces-CLEANED-:-Gene-distribution-per-nuclei-1"
class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/iliana-pap/Alveolar_Mph_mouse_lung_PH/blob/6933769e8e13c56ea87bb35c20b0d3aead2afbf8/docs/figure/MuLung_snRNAseq_preprocessing.Rmd/CLEANED : Gene distribution per nuclei-1.png" target="_blank">6933769</a>
</td>
<td>
iliana-pap
</td>
<td>
2025-05-07
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="genes-detected-per-umi-1" class="section level3">
<h3>genes detected per UMI</h3>
<pre class="r"><code># Add number of genes per UMI for each cell to metadata - measures complexity of dataset 
LungMu_obj$log10GenesPerUMI &lt;- log10(LungMu_obj$nFeature_RNA) / log10(LungMu_obj$nCount_RNA)

# Visualize the overall complexity of the gene expression by visualizing the genes detected per UMI (novelty score)
LungMu_obj@meta.data %&gt;%
    ggplot(aes(x=log10GenesPerUMI, color = dataset, fill=dataset)) +
    scale_fill_manual(values=colDataset) +
    geom_density(alpha = 0.2) +
    scale_x_log10() +
    theme_classic()</code></pre>
<p><img src="figure/MuLung_snRNAseq_preprocessing.Rmd/CLEANED%20:%20gene%20expression%20complexity-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-CLEANED-:-gene-expression-complexity-1">
Past versions of "CLEANED : gene expression complexity-1.png"
</button>
</p>
<div id="fig-no-spaces-CLEANED-:-gene-expression-complexity-1"
class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/iliana-pap/Alveolar_Mph_mouse_lung_PH/blob/6933769e8e13c56ea87bb35c20b0d3aead2afbf8/docs/figure/MuLung_snRNAseq_preprocessing.Rmd/CLEANED : gene expression complexity-1.png" target="_blank">6933769</a>
</td>
<td>
iliana-pap
</td>
<td>
2025-05-07
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div
id="compute-and-visualize-number-of-nuclei-per-sample-via-a-pie-chart-1"
class="section level2">
<h2>Compute and visualize number of nuclei per sample via a pie
chart</h2>
<pre class="r"><code>cells_per_dataset &lt;- table(LungMu_obj$dataset)
cells_per_datasetDF &lt;-  data.frame(
  sample=names(cells_per_dataset) ,  
  cells=as.vector(cells_per_dataset)
  )

# Compute percentages
cells_per_datasetDF$fraction &lt;- cells_per_datasetDF$cells / sum(cells_per_datasetDF$cells)

# Compute the cumulative percentages (top of each rectangle)
cells_per_datasetDF$ymax &lt;- cumsum(cells_per_datasetDF$fraction)

# Compute the bottom of each rectangle
cells_per_datasetDF$ymin &lt;- c(0, head(cells_per_datasetDF$ymax, n=-1))

# Compute label position
cells_per_datasetDF$labelPosition &lt;- (cells_per_datasetDF$ymax + cells_per_datasetDF$ymin) / 2

# Compute a good label
cells_per_datasetDF$label &lt;- paste0(cells_per_datasetDF$patient, &quot;\n value: &quot;, cells_per_datasetDF$nuclei)

# Make the plot
p&lt;- ggplot(cells_per_datasetDF, aes(ymax=ymax, ymin=ymin, xmax=4, xmin=3, fill=sample)) +
  geom_rect() +
  scale_fill_manual(values = colDataset) +
  coord_polar(theta=&quot;y&quot;, start = 0) +
  xlim(c(2, 4)) +
  theme_void() +
  annotate(geom = &#39;text&#39;, x = 2, y = 0, label = as.character(dim(LungMu_obj)[2]), size = 10)
p</code></pre>
<p><img src="figure/MuLung_snRNAseq_preprocessing.Rmd/CLEANED%20:%20number%20of%20nuclei/sample-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-sample-1">
Past versions of sample-1.png
</button>
</p>
<div id="fig-sample-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/iliana-pap/Alveolar_Mph_mouse_lung_PH/blob/6933769e8e13c56ea87bb35c20b0d3aead2afbf8/docs/figure/MuLung_snRNAseq_preprocessing.Rmd/CLEANED : number of nuclei/sample-1.png" target="_blank">6933769</a>
</td>
<td>
iliana-pap
</td>
<td>
2025-05-07
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="save-processed-seurat-object" class="section level2">
<h2>Save processed seurat object</h2>
<pre class="r"><code>saveRDS(LungMu_obj, file=paste0(output_obj_dir, &quot;340821_Mu_Lung_snRNAseq_merged_preprocessed.rds&quot; ))</code></pre>
</div>
<div id="session-information" class="section level2">
<h2>Session Information</h2>
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.3.1 (2023-06-16)
Platform: aarch64-apple-darwin20 (64-bit)
Running under: macOS 15.3.2

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRblas.0.dylib 
LAPACK: /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.11.0

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

time zone: Europe/Zurich
tzcode source: internal

attached base packages:
[1] grid      stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] gridExtra_2.3      scCustomize_1.1.3  viridis_0.6.4      viridisLite_0.4.2 
 [5] pheatmap_1.0.12    ggsci_3.0.0        zeallot_0.1.0      openxlsx_4.2.5.2  
 [9] here_1.0.1         lubridate_1.9.4    forcats_1.0.0      purrr_1.0.2       
[13] readr_2.1.4        tidyr_1.3.1        tibble_3.2.1       ggplot2_3.5.2     
[17] tidyverse_2.0.0    dplyr_1.1.4        stringr_1.5.0      SeuratObject_5.0.2
[21] Seurat_4.3.0.1     patchwork_1.3.0    plyr_1.8.9         workflowr_1.7.1   

loaded via a namespace (and not attached):
  [1] RColorBrewer_1.1-3     shape_1.4.6            rstudioapi_0.15.0     
  [4] jsonlite_1.8.7         magrittr_2.0.3         ggbeeswarm_0.7.2      
  [7] spatstat.utils_3.0-5   farver_2.1.1           rmarkdown_2.24        
 [10] GlobalOptions_0.1.2    fs_1.6.3               vctrs_0.6.5           
 [13] ROCR_1.0-11            spatstat.explore_3.2-1 paletteer_1.6.0       
 [16] janitor_2.2.0          htmltools_0.5.6        sass_0.4.7            
 [19] sctransform_0.3.5      parallelly_1.37.1      KernSmooth_2.23-21    
 [22] bslib_0.5.1            htmlwidgets_1.6.2      ica_1.0-3             
 [25] plotly_4.10.2          zoo_1.8-12             cachem_1.0.8          
 [28] whisker_0.4.1          igraph_1.5.1           mime_0.12             
 [31] lifecycle_1.0.4        pkgconfig_2.0.3        Matrix_1.6-4          
 [34] R6_2.5.1               fastmap_1.1.1          snakecase_0.11.0      
 [37] fitdistrplus_1.1-11    future_1.33.2          shiny_1.7.5           
 [40] digest_0.6.36          colorspace_2.1-0       rematch2_2.1.2        
 [43] ps_1.7.5               rprojroot_2.0.3        tensor_1.5            
 [46] irlba_2.3.5.1          labeling_0.4.2         progressr_0.14.0      
 [49] timechange_0.3.0       fansi_1.0.4            spatstat.sparse_3.0-2 
 [52] httr_1.4.6             polyclip_1.10-4        abind_1.4-5           
 [55] compiler_4.3.1         withr_3.0.2            highr_0.10            
 [58] MASS_7.3-60            tools_4.3.1            vipor_0.4.5           
 [61] lmtest_0.9-40          beeswarm_0.4.0         zip_2.3.0             
 [64] httpuv_1.6.11          future.apply_1.11.2    goftest_1.2-3         
 [67] glue_1.8.0             callr_3.7.3            nlme_3.1-162          
 [70] promises_1.2.1         Rtsne_0.16             getPass_0.2-4         
 [73] cluster_2.1.4          reshape2_1.4.4         generics_0.1.3        
 [76] gtable_0.3.3           spatstat.data_3.0-1    tzdb_0.4.0            
 [79] hms_1.1.3              data.table_1.14.8      sp_2.1-4              
 [82] utf8_1.2.3             spatstat.geom_3.2-4    RcppAnnoy_0.0.21      
 [85] ggrepel_0.9.6          RANN_2.6.1             pillar_1.9.0          
 [88] ggprism_1.0.4          spam_2.10-0            later_1.3.1           
 [91] circlize_0.4.15        splines_4.3.1          lattice_0.21-8        
 [94] survival_3.5-5         deldir_1.0-9           tidyselect_1.2.0      
 [97] miniUI_0.1.1.1         pbapply_1.7-2          knitr_1.43            
[100] git2r_0.36.2           scattermore_1.2        xfun_0.40             
[103] matrixStats_1.0.0      stringi_1.7.12         lazyeval_0.2.2        
[106] yaml_2.3.7             evaluate_0.21          codetools_0.2-19      
[109] cli_3.6.3              uwot_0.1.16            xtable_1.8-4          
[112] reticulate_1.31        munsell_0.5.0          processx_3.8.2        
[115] jquerylib_0.1.4        Rcpp_1.0.13            globals_0.16.3        
[118] spatstat.random_3.1-5  png_0.1-8              ggrastr_1.0.2         
[121] parallel_4.3.1         ellipsis_0.3.2         dotCall64_1.1-1       
[124] listenv_0.9.1          scales_1.3.0           ggridges_0.5.4        
[127] leiden_0.4.3           rlang_1.1.4            cowplot_1.1.1         </code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.3.1 (2023-06-16)
Platform: aarch64-apple-darwin20 (64-bit)
Running under: macOS 15.3.2

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRblas.0.dylib 
LAPACK: /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.11.0

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

time zone: Europe/Zurich
tzcode source: internal

attached base packages:
[1] grid      stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] gridExtra_2.3      scCustomize_1.1.3  viridis_0.6.4      viridisLite_0.4.2 
 [5] pheatmap_1.0.12    ggsci_3.0.0        zeallot_0.1.0      openxlsx_4.2.5.2  
 [9] here_1.0.1         lubridate_1.9.4    forcats_1.0.0      purrr_1.0.2       
[13] readr_2.1.4        tidyr_1.3.1        tibble_3.2.1       ggplot2_3.5.2     
[17] tidyverse_2.0.0    dplyr_1.1.4        stringr_1.5.0      SeuratObject_5.0.2
[21] Seurat_4.3.0.1     patchwork_1.3.0    plyr_1.8.9         workflowr_1.7.1   

loaded via a namespace (and not attached):
  [1] RColorBrewer_1.1-3     shape_1.4.6            rstudioapi_0.15.0     
  [4] jsonlite_1.8.7         magrittr_2.0.3         ggbeeswarm_0.7.2      
  [7] spatstat.utils_3.0-5   farver_2.1.1           rmarkdown_2.24        
 [10] GlobalOptions_0.1.2    fs_1.6.3               vctrs_0.6.5           
 [13] ROCR_1.0-11            spatstat.explore_3.2-1 paletteer_1.6.0       
 [16] janitor_2.2.0          htmltools_0.5.6        sass_0.4.7            
 [19] sctransform_0.3.5      parallelly_1.37.1      KernSmooth_2.23-21    
 [22] bslib_0.5.1            htmlwidgets_1.6.2      ica_1.0-3             
 [25] plotly_4.10.2          zoo_1.8-12             cachem_1.0.8          
 [28] whisker_0.4.1          igraph_1.5.1           mime_0.12             
 [31] lifecycle_1.0.4        pkgconfig_2.0.3        Matrix_1.6-4          
 [34] R6_2.5.1               fastmap_1.1.1          snakecase_0.11.0      
 [37] fitdistrplus_1.1-11    future_1.33.2          shiny_1.7.5           
 [40] digest_0.6.36          colorspace_2.1-0       rematch2_2.1.2        
 [43] ps_1.7.5               rprojroot_2.0.3        tensor_1.5            
 [46] irlba_2.3.5.1          labeling_0.4.2         progressr_0.14.0      
 [49] timechange_0.3.0       fansi_1.0.4            spatstat.sparse_3.0-2 
 [52] httr_1.4.6             polyclip_1.10-4        abind_1.4-5           
 [55] compiler_4.3.1         withr_3.0.2            highr_0.10            
 [58] MASS_7.3-60            tools_4.3.1            vipor_0.4.5           
 [61] lmtest_0.9-40          beeswarm_0.4.0         zip_2.3.0             
 [64] httpuv_1.6.11          future.apply_1.11.2    goftest_1.2-3         
 [67] glue_1.8.0             callr_3.7.3            nlme_3.1-162          
 [70] promises_1.2.1         Rtsne_0.16             getPass_0.2-4         
 [73] cluster_2.1.4          reshape2_1.4.4         generics_0.1.3        
 [76] gtable_0.3.3           spatstat.data_3.0-1    tzdb_0.4.0            
 [79] hms_1.1.3              data.table_1.14.8      sp_2.1-4              
 [82] utf8_1.2.3             spatstat.geom_3.2-4    RcppAnnoy_0.0.21      
 [85] ggrepel_0.9.6          RANN_2.6.1             pillar_1.9.0          
 [88] ggprism_1.0.4          spam_2.10-0            later_1.3.1           
 [91] circlize_0.4.15        splines_4.3.1          lattice_0.21-8        
 [94] survival_3.5-5         deldir_1.0-9           tidyselect_1.2.0      
 [97] miniUI_0.1.1.1         pbapply_1.7-2          knitr_1.43            
[100] git2r_0.36.2           scattermore_1.2        xfun_0.40             
[103] matrixStats_1.0.0      stringi_1.7.12         lazyeval_0.2.2        
[106] yaml_2.3.7             evaluate_0.21          codetools_0.2-19      
[109] cli_3.6.3              uwot_0.1.16            xtable_1.8-4          
[112] reticulate_1.31        munsell_0.5.0          processx_3.8.2        
[115] jquerylib_0.1.4        Rcpp_1.0.13            globals_0.16.3        
[118] spatstat.random_3.1-5  png_0.1-8              ggrastr_1.0.2         
[121] parallel_4.3.1         ellipsis_0.3.2         dotCall64_1.1-1       
[124] listenv_0.9.1          scales_1.3.0           ggridges_0.5.4        
[127] leiden_0.4.3           rlang_1.1.4            cowplot_1.1.1         </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
