<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Iliana Papadopoulou" />

<meta name="date" content="2025-05-08" />

<title>MuLung_snRNAseq_analysis</title>

<script src="site_libs/header-attrs-2.24/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Alveolar_Mph_mouse_lung_PH</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">MuLung_snRNAseq_analysis</h1>
<h4 class="author">Iliana Papadopoulou</h4>
<h4 class="date">2025-05-08</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2025-05-08
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong>
<code>Alveolar_Mph_mouse_lung_PH/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20250507code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20250507)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20250507code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20250507)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomilianapapAlveolarMphmouselungPHtree74ed29d7350edcbb3d075f01fdd99ca7b6aeb6f8targetblank74ed29da">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/iliana-pap/Alveolar_Mph_mouse_lung_PH/tree/74ed29d7350edcbb3d075f01fdd99ca7b6aeb6f8" target="_blank">74ed29d</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomilianapapAlveolarMphmouselungPHtree74ed29d7350edcbb3d075f01fdd99ca7b6aeb6f8targetblank74ed29da"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/iliana-pap/Alveolar_Mph_mouse_lung_PH/tree/74ed29d7350edcbb3d075f01fdd99ca7b6aeb6f8" target="_blank">74ed29d</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    data/Output/

Untracked files:
    Untracked:  analysis/Lung_snRNAseq_functions.R

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown
(<code>analysis/MuLung_snRNAseq_analysis.Rmd</code>) and HTML
(<code>docs/MuLung_snRNAseq_analysis.html</code>) files. If you’ve
configured a remote Git repository (see <code>?wflow_git_remote</code>),
click on the hyperlinks in the table below to view the files as they
were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/iliana-pap/Alveolar_Mph_mouse_lung_PH/blob/74ed29d7350edcbb3d075f01fdd99ca7b6aeb6f8/analysis/MuLung_snRNAseq_analysis.Rmd" target="_blank">74ed29d</a>
</td>
<td>
iliana-pap
</td>
<td>
2025-05-08
</td>
<td>
Perform unbiased graph-based clustering, annotate populations, extract
DE genes, find differences between conditions
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div
id="analysis-of-murine-lung-samples-from-healthy-mice-vs-mice-with-hfpef-in-pulmonary-hypertension-ph"
class="section level1">
<h1>2. Analysis of murine lung samples from healthy mice vs mice with
HFpEF in Pulmonary Hypertension (PH)</h1>
<div id="load-r-libraries" class="section level2">
<h2>Load R libraries</h2>
<pre class="r"><code>suppressPackageStartupMessages({
  library(plyr)
  library(patchwork)
  library(Seurat)
  library(stringr)
  library(dplyr)
  library(tidyverse)
  library(here)
  library(openxlsx)
  library(zeallot)
  library(ggsci)
  library(openxlsx)
  library(pheatmap)
  library(stringr)
  library(viridis)
  library(scCustomize)
  library(stringr)
  library(gridExtra)
  library(grid)
})</code></pre>
</div>
<div id="set-directory" class="section level2">
<h2>Set directory</h2>
<pre class="r"><code>basedir &lt;- here()
source(paste0(basedir,&#39;/analysis/Lung_snRNAseq_functions.R&#39;), echo=FALSE)

# Create directory to save preprocessed seurat objects
output_obj_dir &lt;- paste0(basedir,&quot;/data/Output/&quot;)
dir.create(output_obj_dir)</code></pre>
</div>
<div id="read-preprocessed-murine-lung-seurat-object"
class="section level2">
<h2>Read preprocessed murine lung seurat object</h2>
<pre class="r"><code>LungMu_obj &lt;- readRDS(file= paste0(output_obj_dir, &quot;340821_Mu_Lung_snRNAseq_merged_preprocessed.rds&quot; ))</code></pre>
</div>
<div id="color-vectors---define-color-palete" class="section level2">
<h2>Color vectors - Define color palete</h2>
<pre class="r"><code>cols&lt;- pal_igv()(51)
names(cols) &lt;- c(0:50)

colDataset &lt;- c(&quot;#FF7F0EFF&quot;,&quot;#17BECFFF&quot;, &quot;#BB0021FF&quot;, &quot;#00CC33FF&quot;, &quot;#3B4992FF&quot;, &quot;#4775FFFF&quot;, &quot;#990080FF&quot;, &quot;#AEC7E8FF&quot;,&quot;#FFBB78FF&quot;,&quot;#1B1919FF&quot;)
names(colDataset) &lt;- unique(LungMu_obj$dataset)

colCond &lt;- c(&quot;#631879FF&quot;,&quot;#FF7F0EFF&quot;) #&quot;#BCBD22FF&quot;
names(colCond) &lt;- unique(LungMu_obj$condition)</code></pre>
</div>
<div id="preprocessing" class="section level2">
<h2>Preprocessing</h2>
<pre class="r"><code>resolution &lt;- c(0.1, 0.15,0.2, 0.25,0.3, 0.4, 0.5, 0.6, 0.8, 1., 1.2, 1.5, 1.7, 2, 2.2, 2.4, 2.6)
LungMu_obj  &lt;- preprocessing(LungMu_obj,resolution)</code></pre>
<pre><code>Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 15822
Number of edges: 630740

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.9800
Number of communities: 9
Elapsed time: 1 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 15822
Number of edges: 630740

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.9727
Number of communities: 12
Elapsed time: 1 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 15822
Number of edges: 630740

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.9661
Number of communities: 13
Elapsed time: 1 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 15822
Number of edges: 630740

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.9603
Number of communities: 13
Elapsed time: 1 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 15822
Number of edges: 630740

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.9545
Number of communities: 14
Elapsed time: 1 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 15822
Number of edges: 630740

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.9449
Number of communities: 16
Elapsed time: 1 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 15822
Number of edges: 630740

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.9358
Number of communities: 17
Elapsed time: 1 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 15822
Number of edges: 630740

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.9275
Number of communities: 19
Elapsed time: 1 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 15822
Number of edges: 630740

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.9145
Number of communities: 24
Elapsed time: 1 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 15822
Number of edges: 630740

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.9041
Number of communities: 28
Elapsed time: 1 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 15822
Number of edges: 630740

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.8937
Number of communities: 29
Elapsed time: 1 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 15822
Number of edges: 630740

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.8792
Number of communities: 32
Elapsed time: 1 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 15822
Number of edges: 630740

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.8707
Number of communities: 34
Elapsed time: 1 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 15822
Number of edges: 630740

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.8583
Number of communities: 36
Elapsed time: 1 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 15822
Number of edges: 630740

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.8503
Number of communities: 38
Elapsed time: 1 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 15822
Number of edges: 630740

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.8430
Number of communities: 39
Elapsed time: 1 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 15822
Number of edges: 630740

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.8360
Number of communities: 42
Elapsed time: 1 seconds</code></pre>
</div>
<div id="cell-type-annotation" class="section level2">
<h2>Cell type annotation</h2>
<pre class="r"><code>LungMu_obj$label &lt;- &quot;other&quot;
LungMu_obj$label[which(LungMu_obj$RNA_snn_res.0.15 %in% c(&quot;0&quot;))] &lt;- &quot;AT2 Epithelial&quot;
LungMu_obj$label[which(LungMu_obj$RNA_snn_res.0.15 %in% c(&quot;1&quot;))] &lt;-  &quot;AT1 Epithelial&quot;
LungMu_obj$label[which(LungMu_obj$RNA_snn_res.0.15 %in% c(&quot;2&quot;))] &lt;- &quot;Bas/SMG duct Epithelial 1&quot;
LungMu_obj$label[which(LungMu_obj$RNA_snn_res.0.15 %in% c(&quot;3&quot;))] &lt;- &quot;Fibroblasts&quot;
LungMu_obj$label[which(LungMu_obj$RNA_snn_res.0.15 %in% c(&quot;4&quot;))] &lt;- &quot;Arterial/Cap pulmonary EC&quot;
LungMu_obj$label[which(LungMu_obj$RNA_snn_res.0.15 %in% c(&quot;5&quot;))] &lt;- &quot;Alveolar Mph&quot;
LungMu_obj$label[which(LungMu_obj$RNA_snn_res.0.15 %in% c(&quot;6&quot;))] &lt;- &quot;Bas/SMG duct Epithelial 2&quot;
LungMu_obj$label[which(LungMu_obj$RNA_snn_res.0.15 %in% c(&quot;7&quot;))] &lt;- &quot;Cil Epithelial&quot;
LungMu_obj$label[which(LungMu_obj$RNA_snn_res.0.15 %in% c(&quot;8&quot;))] &lt;-  &quot;Immune cells&quot;
LungMu_obj$label[which(LungMu_obj$RNA_snn_res.0.15 %in% c(&quot;9&quot;))] &lt;- &quot;aCap pulmonary EC&quot;
LungMu_obj$label[which(LungMu_obj$RNA_snn_res.0.15 %in% c(&quot;10&quot;))] &lt;-  &quot;Mesothelial&quot;
LungMu_obj$label[which(LungMu_obj$RNA_snn_res.0.15 %in% c(&quot;11&quot;))] &lt;-  &quot;VSMC/Pericytes&quot;

colLab &lt;- pal_igv()(length(unique(LungMu_obj$label)))
names(colLab) &lt;- unique(LungMu_obj$label)

colLab[&quot;AT2 Epithelial&quot;] &lt;-&quot;#92C5DE&quot;
colLab[&quot;Bas/SMG duct Epithelial 1&quot;] &lt;-&quot;#F4A582&quot; 
colLab[&quot;Bas/SMG duct Epithelial 2&quot;] &lt;-&quot;#FDDBC7&quot; 
colLab[&quot;AT1 Epithelial&quot;] &lt;-&quot;#9467BDFF&quot;
colLab[&quot;Fibroblasts&quot;] &lt;- &quot;#FF1463FF&quot;
colLab[&quot;Arterial/Cap pulmonary EC&quot;] &lt;-&quot;#990080FF&quot;
colLab[&quot;Alveolar Mph&quot;] &lt;-&quot;#00CC33FF&quot;
colLab[&quot;Cil Epithelial&quot;] &lt;- &quot;#009966FF&quot;
colLab[&quot;Immune cells&quot;] &lt;- &quot;#053061&quot;
colLab[&quot;aCap pulmonary EC&quot;] &lt;- &quot;#0099CCFF&quot;
colLab[&quot;Mesothelial&quot;] &lt;- &quot;#67001F&quot;
colLab[&quot;VSMC/Pericytes&quot;] &lt;- &quot;#E7C76FFF&quot;</code></pre>
</div>
<div id="sample-visualization-and-clustering"
class="section level2 tabset">
<h2 class="tabset">Sample Visualization and Clustering</h2>
<div id="figure-2a---cluster-annotation" class="section level3">
<h3>Figure 2A - cluster annotation</h3>
<pre class="r"><code>colLab_reorder &lt;- colLab[order(factor(names(colLab), levels = c(&quot;Arterial/Cap pulmonary EC&quot;,&quot;aCap pulmonary EC&quot;,
                                                                &quot;Bas/SMG duct Epithelial 1&quot;,&quot;Bas/SMG duct Epithelial 2&quot;,&quot;AT1 Epithelial&quot;,&quot;AT2 Epithelial&quot;, &quot;Cil Epithelial&quot;,
                                                                &quot;Fibroblasts&quot;,&quot;VSMC/Pericytes&quot;,&quot;Mesothelial&quot;,
                                                                &quot;Alveolar Mph&quot;, &quot;Immune cells&quot;)))]

DimPlot(LungMu_obj, reduction = &quot;umap&quot;, group.by = &quot;label&quot;, cols = colLab_reorder, order =  names(rev(colLab_reorder)))+
  theme_bw() +
  theme(axis.line = element_line(colour = &quot;black&quot;),
    panel.grid.major = element_blank(),
    panel.border = element_blank(),
    panel.background = element_blank(),
    axis.text = element_blank(), axis.ticks = element_blank(), 
    panel.grid.minor = element_blank()) +
  ggtitle(&quot;&quot;) +
  xlab(&quot;UMAP1&quot;) +
  ylab(&quot;UMAP2&quot;)</code></pre>
<p><img src="figure/MuLung_snRNAseq_analysis.Rmd/Figure%202A%20-%20cluster%20annotation-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="figure-2b---umap-per-condition" class="section level3">
<h3>Figure 2B - Umap per condition</h3>
<pre class="r"><code>DimPlot(LungMu_obj, reduction = &quot;umap&quot;, group.by = paste0(&quot;condition&quot;), cols = colCond)+
  theme_bw() +
  theme( axis.line = element_line(colour = &quot;black&quot;),
    panel.grid.major = element_blank(),
    panel.border = element_blank(),
    panel.background = element_blank(),
    panel.grid.minor = element_blank()) +
  ggtitle(&quot;&quot;) +
  xlab(&quot;UMAP1&quot;) +
  ylab(&quot;UMAP2&quot;)</code></pre>
<p><img src="figure/MuLung_snRNAseq_analysis.Rmd/Umap%20per%20condition-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="umap-per-dataset" class="section level3">
<h3>Umap per dataset</h3>
<pre class="r"><code>DimPlot(LungMu_obj, reduction = &quot;umap&quot;, group.by = paste0(&quot;dataset&quot;), cols = colDataset)+
  theme_bw() +
  theme( axis.line = element_line(colour = &quot;black&quot;),
    panel.grid.major = element_blank(),
    panel.border = element_blank(),
    panel.background = element_blank(),
    axis.text = element_blank(), axis.ticks = element_blank(), 
    panel.grid.minor = element_blank()) +
  ggtitle(&quot;&quot;) +
  xlab(&quot;UMAP1&quot;) +
  ylab(&quot;UMAP2&quot;)</code></pre>
<p><img src="figure/MuLung_snRNAseq_analysis.Rmd/Umap%20per%20dataset-1.png" width="864" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="compute-de-genes-for-all-clusters" class="section level2">
<h2>Compute DE genes for all clusters</h2>
<pre class="r"><code>Idents(LungMu_obj) &lt;- &quot;label&quot;
de.markers &lt;- FindAllMarkers(LungMu_obj, logfc.threshold = 0.25,min.pct = 0.1,only.pos = TRUE)

# plot DE genes top 10 avg logFC
analysis_type &lt;- &quot;total_marker_genes&quot;

genes &lt;- data.frame(gene=rownames(LungMu_obj)) %&gt;% 
    mutate(geneID=gsub(&quot;^.*\\.&quot;, &quot;&quot;, gene)) 

Selgenes &lt;- c(&quot;Hey1&quot;, &quot;Cxcl12&quot;, &quot;Notch4&quot;, &quot;Edn1&quot;,
              &quot;Ednrb&quot;,
              &quot;Muc5b&quot;,&quot;Krt19&quot;,&quot;Dlk2&quot;, 
              &quot;Klf5&quot;,&quot;Hes1&quot;,&quot;Mgst1&quot;, &quot;Klf15&quot;,
              &quot;Clic5&quot;, &quot;Spock2&quot;, &quot;Ager&quot;, &quot;Hopx&quot;, 
             &quot;Sftpb&quot;, &quot;Sftpa1&quot;, &quot;Etv5&quot;, &quot;Acsl4&quot;,
              &quot;Foxj1&quot;, &quot;Caps2&quot;, &quot;Ccdc151&quot;, &quot;Dlec1&quot;,
             &quot;Col1a2&quot; , &quot;Col1a1&quot;, &quot;Pdgfra&quot;,
             &quot;Myh11&quot;, &quot;Tagln&quot;, &quot;Pdgfrb&quot;, &quot;Carmn&quot;,
              &quot;Muc16&quot;, &quot;Msln&quot;, &quot;Wt1&quot;, &quot;Bnc1&quot;,
              &quot;Itgax&quot;, &quot;Marco&quot;, &quot;Siglecf&quot;,&quot;Chil3&quot;,
              &quot;F13a1&quot;,&quot;Ighm&quot;, &quot;Il7r&quot;, &quot;Klrd1&quot;
              )

Selgenes_df &lt;- filter(genes, geneID %in% Selgenes)
colnames(Selgenes_df) &lt;- c(&#39;geneID&#39;,&#39;gene&#39;)
# order by Selgenes
Selgenes_df &lt;-Selgenes_df[match(Selgenes, Selgenes_df$gene),]</code></pre>
</div>
<div id="explore-cell-populations" class="section level2 tabset">
<h2 class="tabset">Explore cell populations</h2>
<div id="figure-s2a---visualize-de-genes-over-all-cell-populations"
class="section level3">
<h3>Figure S2A - Visualize DE genes over all cell populations</h3>
<pre class="r"><code>Idents(LungMu_obj) &lt;- LungMu_obj$label
pOut &lt;- avgHeatmapTotalHeatmap(seurat = LungMu_obj, selGenes = Selgenes_df,
                  colVecIdent = colLab_reorder, 
                  ordVec=levels(LungMu_obj),
                  gapVecR=c(4,5,8, 12,16,20, 24, 27, 31, 35, 39 ), gapVecC=NULL,cc=F,
                  cr=F, condCol=F,
                  path = GeneExprDir,
                  analysis_type = analysis_type,
                  len = 0)</code></pre>
<p><img src="figure/MuLung_snRNAseq_analysis.Rmd/Visualize%20DE%20genes%20over%20all%20cell%20populations-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="figure-s2b---differences-per-condition" class="section level3">
<h3>Figure S2B - Differences per condition</h3>
<pre class="r"><code>DimPlot(LungMu_obj,split.by = &quot;condition&quot;,cols = colLab, group.by = &quot;label&quot;) </code></pre>
<p><img src="figure/MuLung_snRNAseq_analysis.Rmd/Differences%20per%20condition-1.png" width="864" style="display: block; margin: auto;" /></p>
</div>
<div
id="figure-s3c---plot-distribution-of-pulmonary-cells-in-control-mice"
class="section level3">
<h3>Figure S3C - Plot distribution of Pulmonary cells in control
mice</h3>
<pre class="r"><code>percentage_matrix  &lt;- as.data.frame(LungMu_obj@meta.data %&gt;% dplyr::count(condition, label) %&gt;% dplyr::group_by(condition) %&gt;% mutate(cell_percentage= prop.table(n) * 100)) # RNA_snn_res.0.2

distr_df &lt;- percentage_matrix[percentage_matrix$condition == &quot;Control&quot;,]
distr_df$condition &lt;- NULL
names(distr_df)[names(distr_df) == &#39;label&#39;] &lt;- &#39;sample&#39;
names(distr_df)[names(distr_df) == &#39;n&#39;] &lt;- &#39;cells&#39;
names(distr_df)[names(distr_df) == &#39;cell_percentage&#39;] &lt;- &#39;fraction&#39;
distr_df$cell_percentage &lt;- distr_df$fraction / 100

cells_per_datasetDF &lt;- distr_df
# Compute the cumulative percentages (top of each rectangle)
cells_per_datasetDF$ymax &lt;- cumsum(cells_per_datasetDF$fraction)

# Compute the bottom of each rectangle
cells_per_datasetDF$ymin &lt;- c(0, head(cells_per_datasetDF$ymax, n=-1))

# Compute label position
cells_per_datasetDF$labelPosition &lt;- (cells_per_datasetDF$ymax + cells_per_datasetDF$ymin) / 2

# Compute a good label
cells_per_datasetDF$label &lt;- paste0(cells_per_datasetDF$patient, &quot;\n value: &quot;, cells_per_datasetDF$nuclei)

# Make the plot
p&lt;- ggplot(cells_per_datasetDF, aes(ymax=ymax, ymin=ymin, xmax=4, xmin=3, fill=sample)) +
  geom_rect() +
  scale_fill_manual(values = colLab_reorder) +
  coord_polar(theta=&quot;y&quot;, start = 0) +
  xlim(c(2, 4)) +
  theme_void() +
  annotate(geom = &#39;text&#39;, x = 2, y = 0, label = as.character(dim(subset(LungMu_obj,condition == &quot;Control&quot; ))[2]), size = 10)
p</code></pre>
<p><img src="figure/MuLung_snRNAseq_analysis.Rmd/distribution%20of%20Pulmonary%20cells%20in%20control%20%20mice-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="figure-s3c---plot-distribution-of-pulmonary-cells-in-hfd-mice"
class="section level3">
<h3>Figure S3C - Plot distribution of Pulmonary cells in HFD mice</h3>
<pre class="r"><code>percentage_matrix  &lt;- as.data.frame(LungMu_obj@meta.data %&gt;% dplyr::count(condition, label) %&gt;% dplyr::group_by(condition) %&gt;% mutate(cell_percentage= prop.table(n) * 100)) 

distr_df &lt;- percentage_matrix[percentage_matrix$condition == &quot;PH-cHFpEF&quot;,]
distr_df$condition &lt;- NULL
names(distr_df)[names(distr_df) == &#39;label&#39;] &lt;- &#39;sample&#39;
names(distr_df)[names(distr_df) == &#39;n&#39;] &lt;- &#39;cells&#39;
names(distr_df)[names(distr_df) == &#39;cell_percentage&#39;] &lt;- &#39;fraction&#39;
distr_df$cell_percentage &lt;- distr_df$fraction / 100

cells_per_datasetDF &lt;- distr_df
# Compute the cumulative percentages (top of each rectangle)
cells_per_datasetDF$ymax &lt;- cumsum(cells_per_datasetDF$fraction)

# Compute the bottom of each rectangle
cells_per_datasetDF$ymin &lt;- c(0, head(cells_per_datasetDF$ymax, n=-1))

# Compute label position
cells_per_datasetDF$labelPosition &lt;- (cells_per_datasetDF$ymax + cells_per_datasetDF$ymin) / 2

# Compute a good label
cells_per_datasetDF$label &lt;- paste0(cells_per_datasetDF$patient, &quot;\n value: &quot;, cells_per_datasetDF$nuclei)

# Make the plot
p&lt;- ggplot(cells_per_datasetDF, aes(ymax=ymax, ymin=ymin, xmax=4, xmin=3, fill=sample)) +
  geom_rect() +
  scale_fill_manual(values = colLab_reorder) +
  coord_polar(theta=&quot;y&quot;, start = 0) +
  xlim(c(2, 4)) +
  theme_void() +
  annotate(geom = &#39;text&#39;, x = 2, y = 0, label = as.character(dim(subset(LungMu_obj,condition == &quot;PH-cHFpEF&quot; ))[2]), size = 10)
p</code></pre>
<p><img src="figure/MuLung_snRNAseq_analysis.Rmd/distribution%20of%20Pulmonary%20cells%20in%20HFD%20%20mice-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="stacked-barplot-cell-type-abundance-per-sample"
class="section level3">
<h3>Stacked Barplot : Cell type abundance per sample</h3>
<pre class="r"><code># Run successfully
percentage_matrix  &lt;- as.data.frame(LungMu_obj@meta.data %&gt;% count(dataset, label) %&gt;% group_by(dataset) %&gt;% mutate(cell_percentage= prop.table(n) * 100))

# Fix order of celltypes
percentage_matrix$label  &lt;- factor (percentage_matrix$label ,
          levels = names(colLab_reorder),
          ordered = TRUE)

p&lt;- ggplot(percentage_matrix,    # Calculate percent within each dataset
       aes(dataset, cell_percentage, fill=label)) +
  geom_bar(stat=&quot;identity&quot;) + 
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
        axis.line = element_line(), # colour = &quot;black&quot;
    panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(),
    panel.border = element_blank(),
    panel.background = element_blank()) +
  geom_density() + # color = &quot;black&quot;
  labs(y= &quot;Cell type (%)&quot; ) +
  scale_fill_manual(values = colLab_reorder)
p</code></pre>
<p><img src="figure/MuLung_snRNAseq_analysis.Rmd/Cell%20type%20abundance%20per%20sample-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
</div>
<div
id="classify-cell-types-into-endothelial-epithelial-immune-and-stroma-cells"
class="section level2">
<h2>Classify cell types into endothelial, epithelial, immune and stroma
cells</h2>
<pre class="r"><code>LungMu_obj$type &lt;- &quot;Other&quot;
LungMu_obj$type[which(LungMu_obj$label %in% c(&quot;Arterial/Cap pulmonary EC&quot;, &quot;aCap pulmonary EC&quot;))] &lt;- &quot;Endothelial&quot;
LungMu_obj$type[which(LungMu_obj$label %in% c(&quot;Bas/SMG duct Epithelial 1&quot;, &quot;Bas/SMG duct Epithelial 2&quot;,&quot;AT2 Epithelial&quot;, &quot;AT1 Epithelial&quot;, &quot;Cil Epithelial&quot;))] &lt;- &quot;Epithelial&quot;
LungMu_obj$type[which(LungMu_obj$label %in% c(&quot;Immune cells&quot;, &quot;Alveolar Mph&quot;))] &lt;- &quot;Immune&quot;
LungMu_obj$type[which(LungMu_obj$label %in% c(&quot;Fibroblasts&quot;, &quot;Mesothelial&quot;, &quot;VSMC/Pericytes&quot;))] &lt;- &quot;Stroma&quot;

colType &lt;- c(&quot;#374E55FF&quot;, &quot;#D6604D&quot;, &quot;#79AF97FF&quot;, &quot;#0099CCFF&quot;)
names(colType) &lt;- unique(LungMu_obj$type)

DimPlot(LungMu_obj, reduction = &quot;umap&quot;, group.by = &quot;type&quot;, cols = colType)+
  theme_bw() +
  theme(axis.line = element_line(colour = &quot;black&quot;),
    panel.grid.major = element_blank(),
    panel.border = element_blank(),
    panel.background = element_blank(),
    axis.text = element_blank(), axis.ticks = element_blank(), 
    panel.grid.minor = element_blank()) +
  ggtitle(&quot;&quot;) +
  xlab(&quot;UMAP1&quot;) +
  ylab(&quot;UMAP2&quot;)</code></pre>
<p><img src="figure/MuLung_snRNAseq_analysis.Rmd/plot%20cells%20per%20endothelial,%20epithelial,%20imuune%20and%20stroma%20cells-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="find-differences-per-condition" class="section level2">
<h2>Find differences per condition</h2>
<pre class="r"><code>for (cellType in unique(LungMu_obj$label)){
  LungMu_obj_sub &lt;- subset(LungMu_obj, label == cellType)
  Idents(LungMu_obj_sub) &lt;- &quot;condition&quot;
  de.markers &lt;- FindAllMarkers(LungMu_obj_sub, logfc.threshold = 0.25,min.pct = 0.1,only.pos = TRUE)
}

LungMu_obj_sub &lt;- subset(LungMu_obj, label == &quot;Bas/SMG duct Epithelial 1&quot;)
Idents(LungMu_obj_sub) &lt;- &quot;condition&quot;
de.markers &lt;- FindAllMarkers(LungMu_obj_sub, logfc.threshold = 0.25,min.pct = 0.1,only.pos = TRUE)

LungMu_obj_sub &lt;- subset(LungMu_obj, label == &quot;Bas/SMG duct Epithelial 2&quot;)
Idents(LungMu_obj_sub) &lt;- &quot;condition&quot;
de.markers &lt;- FindAllMarkers(LungMu_obj_sub, logfc.threshold = 0.25,min.pct = 0.1,only.pos = TRUE)

LungMu_obj_sub &lt;- subset(LungMu_obj, label == &quot;VSMC/Pericytes&quot;)
Idents(LungMu_obj_sub) &lt;- &quot;condition&quot;
de.markers &lt;- FindAllMarkers(LungMu_obj_sub, logfc.threshold = 0.25,min.pct = 0.1,only.pos = TRUE)

LungMu_obj_sub &lt;- subset(LungMu_obj, label == &quot;Arterial/Cap pulmonary EC&quot;)
Idents(LungMu_obj_sub) &lt;- &quot;condition&quot;
de.markers &lt;- FindAllMarkers(LungMu_obj_sub, logfc.threshold = 0.25,min.pct = 0.1,only.pos = TRUE)</code></pre>
</div>
<div id="compute-gene-per-condition-in-each-cell-type"
class="section level2">
<h2>Compute gene per condition in each cell type</h2>
<pre class="r"><code>analysis_type &lt;- &quot;marker_genes_all_among_conditions&quot;
type &lt;- &quot;All_cells&quot;
Selgenes &lt;- c(
              &quot;Epb41&quot;, &quot;Mcf2l&quot;,&quot;Mcam&quot;,&quot;Glp1r&quot;, &quot;Sox11&quot;, &quot;Samd5&quot;, &quot;Vwf&quot;, &quot;Cd93&quot;,
              &quot;Plod1&quot;, &quot;Rin3&quot;, &quot;Tmx3&quot;, &quot;Rock2&quot;, &quot;Csrnp1&quot;, &quot;Sema6a&quot;, &quot;Mapt&quot;, &quot;Arap2&quot;, &quot;Fmo1&quot;, &quot;Meis1&quot;, &quot;Gnaq&quot;,
              &quot;Muc5b&quot;,&quot;Pim3&quot;, &quot;Depp1&quot;,&quot;Nrarp&quot;,&quot;Tob2&quot;, &quot;Nupr1&quot;, &quot;Nectin2&quot;,
              &quot;Rassf9&quot;,  &quot;Reep6&quot;, &quot;Cstf3&quot;, &quot;Utp11&quot;,&quot;Fdx1&quot;, &quot;Aldh3a2&quot;,
              &quot;Tpcn2&quot;, &quot;Bcar3&quot;,  &quot;Pxdc1&quot;, &quot;Akap5&quot;, &quot;Agrn&quot;,&quot;Nt5e&quot;,&quot;Prkag3&quot;,&quot;Ager&quot;, &quot;Dlg4&quot;, &quot;Gprc5a&quot;, &quot;Ankrd1&quot;,
              &quot;Ctsh&quot;,  &quot;Lrrk2&quot;, &quot;Pmvk&quot;, &quot;Slc34a2&quot;,  &quot;Elovl1&quot;, &quot;Lpcat1&quot;,
              &quot;Hsph1&quot;, &quot;Ahsa2&quot;, &quot;Tuba1a&quot;,&quot;Mt1&quot;,
              &quot;Adamtsl2&quot;, &quot;Dpep1&quot;,&quot;Rgs2&quot;, &quot;Plekha6&quot;,&quot;Atp1a2&quot;, &quot;Slc7a10&quot;, &quot;Inmt&quot;,
              &quot;Tagln&quot;,&quot;Myh11&quot;,  &quot;Pdgfrb&quot;, &quot;Carmn&quot;,
              &quot;Timp2&quot;,&quot;Cys1&quot;,&quot;Gas6&quot;, &quot;Gbp3&quot; ,
              &quot;Fkbp5&quot;, &quot;Tfec&quot;, &quot;Ppt1&quot;, &quot;Ap3d1&quot;, &quot;Fcho2&quot;,&quot;Fuca1&quot;, &quot;Chil3&quot;,  &quot;Vim&quot;, &quot;Mmp19&quot;, &quot;Cebpb&quot;,&quot;Marco&quot;,&quot;Ddb1&quot;,&quot;Sgk1&quot;, &quot;Colgalt1&quot;,&quot;Galnt3&quot;,&quot;Cd44&quot;, &quot;Cotl1&quot;,&quot;Plk3&quot;,
              &quot;Rell1&quot;, &quot;Ms4a6c&quot;, &quot;Tbc1d8&quot;,&quot;Cd83&quot;, &quot;Cd226&quot;,&quot;Got1&quot;,&quot;Pmaip1&quot;,&quot;Ighm&quot;,&quot;Stap1&quot;
             )

Idents(LungMu_obj) &lt;- LungMu_obj$condition
Mu_CD &lt;- subset(LungMu_obj, condition  == &quot;Control&quot; )
Mu_HFD &lt;- subset(LungMu_obj, condition  == &quot;PH-cHFpEF&quot;)
 
avg_matrix_CD&lt;-compute_avg_gene_expression_matrix(Mu_CD,Selgenes,names(colLab_reorder))
avg_matrix_HFD&lt;-compute_avg_gene_expression_matrix(Mu_HFD,Selgenes,names(colLab_reorder))</code></pre>
</div>
<div id="plots-gene-per-condition-in-each-cell-type"
class="section level2">
<h2>Plots gene per condition in each cell type</h2>
<pre class="r"><code>avgHeatmap2(Selgenes,avg_matrix_CD, avg_matrix_HFD,
                      colVecIdent = colLab_reorder,colVecCond = colCond,
                      gapVecR=NULL, gapVecC=NULL,cc=T,
                      cr=F, condCol=F,
                      type = type, folder_name =GeneExprDir)</code></pre>
<p><img src="figure/MuLung_snRNAseq_analysis.Rmd/gene%20per%20condition%20in%20each%20cell%20type-1.png" width="864" style="display: block; margin: auto;" /></p>
</div>
<div id="save-annotated-object" class="section level2">
<h2>Save annotated object</h2>
<pre class="r"><code>saveRDS(LungMu_obj, file= paste0(output_obj_dir, &quot;340821_Mu_Lung_snRNAseq_merged_annotated.rds&quot; ))</code></pre>
</div>
<div id="session-information" class="section level2">
<h2>Session Information</h2>
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.3.1 (2023-06-16)
Platform: aarch64-apple-darwin20 (64-bit)
Running under: macOS 15.3.2

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRblas.0.dylib 
LAPACK: /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.11.0

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

time zone: Europe/Zurich
tzcode source: internal

attached base packages:
[1] grid      stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] gridExtra_2.3      scCustomize_1.1.3  viridis_0.6.4      viridisLite_0.4.2 
 [5] pheatmap_1.0.12    ggsci_3.0.0        zeallot_0.1.0      openxlsx_4.2.5.2  
 [9] here_1.0.1         lubridate_1.9.4    forcats_1.0.0      purrr_1.0.2       
[13] readr_2.1.4        tidyr_1.3.1        tibble_3.2.1       ggplot2_3.5.2     
[17] tidyverse_2.0.0    dplyr_1.1.4        stringr_1.5.0      SeuratObject_5.0.2
[21] Seurat_4.3.0.1     patchwork_1.3.0    plyr_1.8.9         workflowr_1.7.1   

loaded via a namespace (and not attached):
  [1] RColorBrewer_1.1-3     shape_1.4.6            rstudioapi_0.15.0     
  [4] jsonlite_1.8.7         magrittr_2.0.3         ggbeeswarm_0.7.2      
  [7] spatstat.utils_3.0-5   farver_2.1.1           rmarkdown_2.24        
 [10] GlobalOptions_0.1.2    fs_1.6.3               vctrs_0.6.5           
 [13] ROCR_1.0-11            spatstat.explore_3.2-1 paletteer_1.6.0       
 [16] janitor_2.2.0          htmltools_0.5.6        sass_0.4.7            
 [19] sctransform_0.3.5      parallelly_1.37.1      KernSmooth_2.23-21    
 [22] bslib_0.5.1            htmlwidgets_1.6.2      ica_1.0-3             
 [25] plotly_4.10.2          zoo_1.8-12             cachem_1.0.8          
 [28] whisker_0.4.1          igraph_1.5.1           mime_0.12             
 [31] lifecycle_1.0.4        pkgconfig_2.0.3        Matrix_1.6-4          
 [34] R6_2.5.1               fastmap_1.1.1          snakecase_0.11.0      
 [37] fitdistrplus_1.1-11    future_1.33.2          shiny_1.7.5           
 [40] digest_0.6.36          colorspace_2.1-0       rematch2_2.1.2        
 [43] ps_1.7.5               rprojroot_2.0.3        tensor_1.5            
 [46] irlba_2.3.5.1          labeling_0.4.2         progressr_0.14.0      
 [49] timechange_0.3.0       fansi_1.0.4            spatstat.sparse_3.0-2 
 [52] httr_1.4.6             polyclip_1.10-4        abind_1.4-5           
 [55] compiler_4.3.1         withr_3.0.2            highr_0.10            
 [58] MASS_7.3-60            tools_4.3.1            vipor_0.4.5           
 [61] lmtest_0.9-40          beeswarm_0.4.0         zip_2.3.0             
 [64] httpuv_1.6.11          future.apply_1.11.2    goftest_1.2-3         
 [67] glue_1.8.0             callr_3.7.3            nlme_3.1-162          
 [70] promises_1.2.1         Rtsne_0.16             getPass_0.2-4         
 [73] cluster_2.1.4          reshape2_1.4.4         generics_0.1.3        
 [76] gtable_0.3.3           spatstat.data_3.0-1    tzdb_0.4.0            
 [79] hms_1.1.3              data.table_1.14.8      sp_2.1-4              
 [82] utf8_1.2.3             spatstat.geom_3.2-4    RcppAnnoy_0.0.21      
 [85] ggrepel_0.9.6          RANN_2.6.1             pillar_1.9.0          
 [88] limma_3.56.2           ggprism_1.0.4          spam_2.10-0           
 [91] later_1.3.1            circlize_0.4.15        splines_4.3.1         
 [94] lattice_0.21-8         survival_3.5-5         deldir_1.0-9          
 [97] tidyselect_1.2.0       miniUI_0.1.1.1         pbapply_1.7-2         
[100] knitr_1.43             git2r_0.36.2           scattermore_1.2       
[103] xfun_0.40              matrixStats_1.0.0      stringi_1.7.12        
[106] lazyeval_0.2.2         yaml_2.3.7             evaluate_0.21         
[109] codetools_0.2-19       cli_3.6.3              uwot_0.1.16           
[112] xtable_1.8-4           reticulate_1.31        munsell_0.5.0         
[115] processx_3.8.2         jquerylib_0.1.4        Rcpp_1.0.13           
[118] globals_0.16.3         spatstat.random_3.1-5  png_0.1-8             
[121] ggrastr_1.0.2          parallel_4.3.1         ellipsis_0.3.2        
[124] dotCall64_1.1-1        listenv_0.9.1          scales_1.3.0          
[127] ggridges_0.5.4         leiden_0.4.3           rlang_1.1.4           
[130] cowplot_1.1.1         </code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.3.1 (2023-06-16)
Platform: aarch64-apple-darwin20 (64-bit)
Running under: macOS 15.3.2

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRblas.0.dylib 
LAPACK: /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.11.0

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

time zone: Europe/Zurich
tzcode source: internal

attached base packages:
[1] grid      stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] gridExtra_2.3      scCustomize_1.1.3  viridis_0.6.4      viridisLite_0.4.2 
 [5] pheatmap_1.0.12    ggsci_3.0.0        zeallot_0.1.0      openxlsx_4.2.5.2  
 [9] here_1.0.1         lubridate_1.9.4    forcats_1.0.0      purrr_1.0.2       
[13] readr_2.1.4        tidyr_1.3.1        tibble_3.2.1       ggplot2_3.5.2     
[17] tidyverse_2.0.0    dplyr_1.1.4        stringr_1.5.0      SeuratObject_5.0.2
[21] Seurat_4.3.0.1     patchwork_1.3.0    plyr_1.8.9         workflowr_1.7.1   

loaded via a namespace (and not attached):
  [1] RColorBrewer_1.1-3     shape_1.4.6            rstudioapi_0.15.0     
  [4] jsonlite_1.8.7         magrittr_2.0.3         ggbeeswarm_0.7.2      
  [7] spatstat.utils_3.0-5   farver_2.1.1           rmarkdown_2.24        
 [10] GlobalOptions_0.1.2    fs_1.6.3               vctrs_0.6.5           
 [13] ROCR_1.0-11            spatstat.explore_3.2-1 paletteer_1.6.0       
 [16] janitor_2.2.0          htmltools_0.5.6        sass_0.4.7            
 [19] sctransform_0.3.5      parallelly_1.37.1      KernSmooth_2.23-21    
 [22] bslib_0.5.1            htmlwidgets_1.6.2      ica_1.0-3             
 [25] plotly_4.10.2          zoo_1.8-12             cachem_1.0.8          
 [28] whisker_0.4.1          igraph_1.5.1           mime_0.12             
 [31] lifecycle_1.0.4        pkgconfig_2.0.3        Matrix_1.6-4          
 [34] R6_2.5.1               fastmap_1.1.1          snakecase_0.11.0      
 [37] fitdistrplus_1.1-11    future_1.33.2          shiny_1.7.5           
 [40] digest_0.6.36          colorspace_2.1-0       rematch2_2.1.2        
 [43] ps_1.7.5               rprojroot_2.0.3        tensor_1.5            
 [46] irlba_2.3.5.1          labeling_0.4.2         progressr_0.14.0      
 [49] timechange_0.3.0       fansi_1.0.4            spatstat.sparse_3.0-2 
 [52] httr_1.4.6             polyclip_1.10-4        abind_1.4-5           
 [55] compiler_4.3.1         withr_3.0.2            highr_0.10            
 [58] MASS_7.3-60            tools_4.3.1            vipor_0.4.5           
 [61] lmtest_0.9-40          beeswarm_0.4.0         zip_2.3.0             
 [64] httpuv_1.6.11          future.apply_1.11.2    goftest_1.2-3         
 [67] glue_1.8.0             callr_3.7.3            nlme_3.1-162          
 [70] promises_1.2.1         Rtsne_0.16             getPass_0.2-4         
 [73] cluster_2.1.4          reshape2_1.4.4         generics_0.1.3        
 [76] gtable_0.3.3           spatstat.data_3.0-1    tzdb_0.4.0            
 [79] hms_1.1.3              data.table_1.14.8      sp_2.1-4              
 [82] utf8_1.2.3             spatstat.geom_3.2-4    RcppAnnoy_0.0.21      
 [85] ggrepel_0.9.6          RANN_2.6.1             pillar_1.9.0          
 [88] limma_3.56.2           ggprism_1.0.4          spam_2.10-0           
 [91] later_1.3.1            circlize_0.4.15        splines_4.3.1         
 [94] lattice_0.21-8         survival_3.5-5         deldir_1.0-9          
 [97] tidyselect_1.2.0       miniUI_0.1.1.1         pbapply_1.7-2         
[100] knitr_1.43             git2r_0.36.2           scattermore_1.2       
[103] xfun_0.40              matrixStats_1.0.0      stringi_1.7.12        
[106] lazyeval_0.2.2         yaml_2.3.7             evaluate_0.21         
[109] codetools_0.2-19       cli_3.6.3              uwot_0.1.16           
[112] xtable_1.8-4           reticulate_1.31        munsell_0.5.0         
[115] processx_3.8.2         jquerylib_0.1.4        Rcpp_1.0.13           
[118] globals_0.16.3         spatstat.random_3.1-5  png_0.1-8             
[121] ggrastr_1.0.2          parallel_4.3.1         ellipsis_0.3.2        
[124] dotCall64_1.1-1        listenv_0.9.1          scales_1.3.0          
[127] ggridges_0.5.4         leiden_0.4.3           rlang_1.1.4           
[130] cowplot_1.1.1         </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
